<!DOCTYPE html>
<html lang="en"><head>
<script src="chap_design_files/libs/clipboard/clipboard.min.js"></script>
<script src="chap_design_files/libs/quarto-html/tabby.min.js"></script>
<script src="chap_design_files/libs/quarto-html/popper.min.js"></script>
<script src="chap_design_files/libs/quarto-html/tippy.umd.min.js"></script>
<link href="chap_design_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="chap_design_files/libs/quarto-html/quarto-html.min.css" rel="stylesheet" data-mode="light">
<link href="chap_design_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles"><meta charset="utf-8">
  <meta name="generator" content="quarto-1.2.269">

  <meta name="author" content="Susan Holmes, Wolfgang Huber">
  <title>Good scientific practice: Design of High Throughput Experiments and their Analysis</title>
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, minimal-ui">
  <link rel="stylesheet" href="chap_design_files/libs/revealjs/dist/reset.css">
  <link rel="stylesheet" href="chap_design_files/libs/revealjs/dist/reveal.css">
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    div.columns{display: flex; gap: min(4vw, 1.5em);}
    div.column{flex: auto; overflow-x: auto;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    ul.task-list li input[type="checkbox"] {
      width: 0.8em;
      margin: 0 0.8em 0.2em -1.6em;
      vertical-align: middle;
    }
    pre > code.sourceCode { white-space: pre; position: relative; }
    pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
    pre > code.sourceCode > span:empty { height: 1.2em; }
    .sourceCode { overflow: visible; }
    code.sourceCode > span { color: inherit; text-decoration: inherit; }
    div.sourceCode { margin: 1em 0; }
    pre.sourceCode { margin: 0; }
    @media screen {
    div.sourceCode { overflow: auto; }
    }
    @media print {
    pre > code.sourceCode { white-space: pre-wrap; }
    pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
    }
    pre.numberSource code
      { counter-reset: source-line 0; }
    pre.numberSource code > span
      { position: relative; left: -4em; counter-increment: source-line; }
    pre.numberSource code > span > a:first-child::before
      { content: counter(source-line);
        position: relative; left: -1em; text-align: right; vertical-align: baseline;
        border: none; display: inline-block;
        -webkit-touch-callout: none; -webkit-user-select: none;
        -khtml-user-select: none; -moz-user-select: none;
        -ms-user-select: none; user-select: none;
        padding: 0 4px; width: 4em;
        color: #aaaaaa;
      }
    pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
    div.sourceCode
      { color: #003b4f; background-color: #f1f3f5; }
    @media screen {
    pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
    }
    code span { color: #003b4f; } /* Normal */
    code span.al { color: #ad0000; } /* Alert */
    code span.an { color: #5e5e5e; } /* Annotation */
    code span.at { color: #657422; } /* Attribute */
    code span.bn { color: #ad0000; } /* BaseN */
    code span.bu { } /* BuiltIn */
    code span.cf { color: #003b4f; } /* ControlFlow */
    code span.ch { color: #20794d; } /* Char */
    code span.cn { color: #8f5902; } /* Constant */
    code span.co { color: #5e5e5e; } /* Comment */
    code span.cv { color: #5e5e5e; font-style: italic; } /* CommentVar */
    code span.do { color: #5e5e5e; font-style: italic; } /* Documentation */
    code span.dt { color: #ad0000; } /* DataType */
    code span.dv { color: #ad0000; } /* DecVal */
    code span.er { color: #ad0000; } /* Error */
    code span.ex { } /* Extension */
    code span.fl { color: #ad0000; } /* Float */
    code span.fu { color: #4758ab; } /* Function */
    code span.im { color: #00769e; } /* Import */
    code span.in { color: #5e5e5e; } /* Information */
    code span.kw { color: #003b4f; } /* Keyword */
    code span.op { color: #5e5e5e; } /* Operator */
    code span.ot { color: #003b4f; } /* Other */
    code span.pp { color: #ad0000; } /* Preprocessor */
    code span.sc { color: #5e5e5e; } /* SpecialChar */
    code span.ss { color: #20794d; } /* SpecialString */
    code span.st { color: #20794d; } /* String */
    code span.va { color: #111111; } /* Variable */
    code span.vs { color: #20794d; } /* VerbatimString */
    code span.wa { color: #5e5e5e; font-style: italic; } /* Warning */
    div.csl-bib-body { }
    div.csl-entry {
      clear: both;
    }
    .hanging div.csl-entry {
      margin-left:2em;
      text-indent:-2em;
    }
    div.csl-left-margin {
      min-width:2em;
      float:left;
    }
    div.csl-right-inline {
      margin-left:2em;
      padding-left:1em;
    }
    div.csl-indent {
      margin-left: 2em;
    }
  </style>
  <link rel="stylesheet" href="chap_design_files/libs/revealjs/dist/theme/quarto.css" id="theme">
  <link href="chap_design_files/libs/revealjs/plugin/quarto-line-highlight/line-highlight.css" rel="stylesheet">
  <link href="chap_design_files/libs/revealjs/plugin/reveal-menu/menu.css" rel="stylesheet">
  <link href="chap_design_files/libs/revealjs/plugin/reveal-menu/quarto-menu.css" rel="stylesheet">
  <link href="chap_design_files/libs/revealjs/plugin/quarto-support/footer.css" rel="stylesheet">
  <style type="text/css">

  .callout {
    margin-top: 1em;
    margin-bottom: 1em;  
    border-radius: .25rem;
  }

  .callout.callout-style-simple { 
    padding: 0em 0.5em;
    border-left: solid #acacac .3rem;
    border-right: solid 1px silver;
    border-top: solid 1px silver;
    border-bottom: solid 1px silver;
    display: flex;
  }

  .callout.callout-style-default {
    border-left: solid #acacac .3rem;
    border-right: solid 1px silver;
    border-top: solid 1px silver;
    border-bottom: solid 1px silver;
  }

  .callout .callout-body-container {
    flex-grow: 1;
  }

  .callout.callout-style-simple .callout-body {
    font-size: 1rem;
    font-weight: 400;
  }

  .callout.callout-style-default .callout-body {
    font-size: 0.9rem;
    font-weight: 400;
  }

  .callout.callout-captioned.callout-style-simple .callout-body {
    margin-top: 0.2em;
  }

  .callout:not(.callout-captioned) .callout-body {
      display: flex;
  }

  .callout:not(.no-icon).callout-captioned.callout-style-simple .callout-content {
    padding-left: 1.6em;
  }

  .callout.callout-captioned .callout-header {
    padding-top: 0.2em;
    margin-bottom: -0.2em;
  }

  .callout.callout-captioned .callout-caption  p {
    margin-top: 0.5em;
    margin-bottom: 0.5em;
  }
    
  .callout.callout-captioned.callout-style-simple .callout-content  p {
    margin-top: 0;
  }

  .callout.callout-captioned.callout-style-default .callout-content  p {
    margin-top: 0.7em;
  }

  .callout.callout-style-simple div.callout-caption {
    border-bottom: none;
    font-size: .9rem;
    font-weight: 600;
    opacity: 75%;
  }

  .callout.callout-style-default  div.callout-caption {
    border-bottom: none;
    font-weight: 600;
    opacity: 85%;
    font-size: 0.9rem;
    padding-left: 0.5em;
    padding-right: 0.5em;
  }

  .callout.callout-style-default div.callout-content {
    padding-left: 0.5em;
    padding-right: 0.5em;
  }

  .callout.callout-style-simple .callout-icon::before {
    height: 1rem;
    width: 1rem;
    display: inline-block;
    content: "";
    background-repeat: no-repeat;
    background-size: 1rem 1rem;
  }

  .callout.callout-style-default .callout-icon::before {
    height: 0.9rem;
    width: 0.9rem;
    display: inline-block;
    content: "";
    background-repeat: no-repeat;
    background-size: 0.9rem 0.9rem;
  }

  .callout-caption {
    display: flex
  }
    
  .callout-icon::before {
    margin-top: 1rem;
    padding-right: .5rem;
  }

  .callout.no-icon::before {
    display: none !important;
  }

  .callout.callout-captioned .callout-body > .callout-content > :last-child {
    margin-bottom: 0.5rem;
  }

  .callout.callout-captioned .callout-icon::before {
    margin-top: .5rem;
    padding-right: .5rem;
  }

  .callout:not(.callout-captioned) .callout-icon::before {
    margin-top: 1rem;
    padding-right: .5rem;
  }

  /* Callout Types */

  div.callout-note {
    border-left-color: #4582ec !important;
  }

  div.callout-note .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAEU0lEQVRYCcVXTWhcVRQ+586kSUMMxkyaElstCto2SIhitS5Ek8xUKV2poatCcVHtUlFQk8mbaaziwpWgglJwVaquitBOfhQXFlqlzSJpFSpIYyXNjBNiTCck7x2/8/LeNDOZxDuEkgOXe++553zfefee+/OYLOXFk3+1LLrRdiO81yNqZ6K9cG0P3MeFaMIQjXssE8Z1JzLO9ls20MBZX7oG8w9GxB0goaPrW5aNMp1yOZIa7Wv6o2ykpLtmAPs/vrG14Z+6d4jpbSKuhdcSyq9wGMPXjonwmESXrriLzFGOdDBLB8Y6MNYBu0dRokSygMA/mrun8MGFN3behm6VVAwg4WR3i6FvYK1T7MHo9BK7ydH+1uurECoouk5MPRyVSBrBHMYwVobG2aOXM07sWrn5qgB60rc6mcwIDJtQrnrEr44kmy+UO9r0u9O5/YbkS9juQckLed3DyW2XV/qWBBB3ptvI8EUY3I9p/67OW+g967TNr3Sotn3IuVlfMLVnsBwH4fsnebJvyGm5GeIUA3jljERmrv49SizPYuq+z7c2H/jlGC+Ghhupn/hcapqmcudB9jwJ/3jvnvu6vu5lVzF1fXyZuZZ7U8nRmVzytvT+H3kilYvH09mLWrQdwFSsFEsxFVs5fK7A0g8gMZjbif4ACpKbjv7gNGaD8bUrlk8x+KRflttr22JEMRUbTUwwDQScyzPgedQHZT0xnx7ujw2jfVfExwYHwOsDTjLdJ2ebmeQIlJ7neo41s/DrsL3kl+W2lWvAga0tR3zueGr6GL78M3ifH0rGXrBC2aAR8uYcIA5gwV8zIE8onoh8u0Fca/ciF7j1uOzEnqcIm59sEXoGc0+z6+H45V1CvAvHcD7THztu669cnp+L0okAeIc6zjbM/24LgGM1gZk7jnRu1aQWoU9sfUOuhrmtaPIO3YY1KLLWZaEO5TKUbMY5zx8W9UJ6elpLwKXbsaZ4EFl7B4bMtDv0iRipKoDQT2sNQI9b1utXFdYisi+wzZ/ri/1m7QfDgEuvgUUEIJPq3DhX/5DWNqIXDOweC2wvIR90Oq3lDpdMIgD2r0dXvGdsEW5H6x6HLRJYU7C69VefO1x8Gde1ZFSJLfWS1jbCnhtOPxmpfv2LXOA2Xk2tvnwKKPFuZ/oRmwBwqRQDcKNeVQkYcOjtWVBuM/JuYw5b6isojIkYxyYAFn5K7ZBF10fea52y8QltAg6jnMqNHFBmGkQ1j+U43HMi2xMar1Nv0zGsf1s8nUsmUtPOOrbFIR8bHFDMB5zL13Gmr/kGlCkUzedTzzmzsaJXhYawnA3UmARpiYj5ooJZiUoxFRtK3X6pgNPv+IZVPcnwbOl6f+aBaO1CNvPW9n9LmCp01nuSaTRF2YxHqZ8DYQT6WsXT+RD6eUztwYLZ8rM+rcPxamv1VQzFUkzFXvkiVrySGQgJNvXHJAxiU3/NwiC03rSf05VBaPtu/Z7/B8Yn/w7eguloAAAAAElFTkSuQmCC');
  }

  div.callout-note.callout-style-default .callout-caption {
    background-color: #dae6fb
  }

  div.callout-important {
    border-left-color: #d9534f !important;
  }

  div.callout-important .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAEKklEQVRYCcVXTWhcVRS+575MJym48A+hSRFr00ySRQhURRfd2HYjk2SSTokuBCkU2o0LoSKKraKIBTcuFCoidGFD08nkBzdREbpQ1EDNIv8qSGMFUboImMSZd4/f9zJv8ibJMC8xJQfO3HPPPef7zrvvvnvviIkpC9nsw0UttFunbUhpFzFtarSd6WJkStVMw5xyVqYTvkwfzuf/5FgtkVoB0729j1rjXwThS7Vio+Mo6DNnvLfahoZ+i/o32lULuJ3NNiz7q6+pyAUkJaFF6JwaM2lUJlV0MlnQn5aTRbEu0SEqHUa0A4AdiGuB1kFXRfVyg5d87+Dg4DL6m2TLAub60ilj7A1Ec4odSAc8X95sHh7+ZRPCFo6Fnp7HfU/fBng/hi10CjCnWnJjsxvDNxWw0NfV6Rv5GgP3I3jGWXumdTD/3cbEOP2ZbOZp69yniG3FQ9z1jD7bnBu9Fc2tKGC2q+uAJOQHBDRiZX1x36o7fWBs7J9ownbtO+n0/qWkvW7UPIfc37WgT6ZGR++EOJyeQDSb9UB+DZ1G6DdLDzyS+b/kBCYGsYgJbSQHuThGKRcw5xdeQf8YdNHsc6ePXrlSYMBuSIAFTGAtQo+VuALo4BX83N190NWZWbynBjhOHsmNfFWLeL6v+ynsA58zDvvAC8j5PkbOcXCMg2PZFk3q8MjI7WAG/Dp9AwP7jdGBOOQkAvlFUB+irtm16I1Zw9YBcpGTGXYmk3kQIC/Cds55l+iMI3jqhjAuaoe+am2Jw5GT3Nbz3CkE12NavmzN5+erJW7046n/CH1RO/RVa8lBLozXk9uqykkGAyRXLWlLv5jyp4RFsG5vGVzpDLnIjTWgnRy2Rr+tDKvRc7Y8AyZq10jj8DqXdnIRNtFZb+t/ZRtXcDiVnzpqx8mPcDWxgARUqx0W1QB9MeUZiNrV4qP+Ehc+BpNgATsTX8ozYKL2NtFYAHc84fG7ndxUPr+AR/iQSns7uSUufAymwDOb2+NjK27lEFocm/EE2WpyIy/Hi66MWuMKJn8RvxIcj87IM5Vh9663ziW36kR0HNenXuxmfaD8JC7tfKbrhFr7LiZCrMjrzTeGx+PmkosrkNzW94ObzwocJ7A1HokLolY+AvkTiD/q1H0cN48c5EL8Crkttsa/AXQVDmutfyku0E7jShx49XqV3MFK8IryDhYVbj7Sj2P2eBxwcXoe8T8idsKKPRcnZw1b+slFTubwUwhktrfnAt7J++jwQtLZcm3sr9LQrjRzz6cfMv9aLvgmnAGvpoaGLxM4mAEaLV7iAzQ3oU0IvD5x9ix3yF2RAAuYAOO2f7PEFWCXZ4C9Pb2UsgDeVnFSpbFK7/IWu7TPTvBqzbGdCHOJQSxiEjt6IyZmxQyEJHv6xyQsYk//moVFsN2zP6fRImjfq7/n/wFDguUQFNEwugAAAABJRU5ErkJggg==');
  }

  div.callout-important.callout-style-default .callout-caption {
    background-color: #f7dddc
  }

  div.callout-warning {
    border-left-color: #f0ad4e !important;
  }

  div.callout-warning .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAAETklEQVRYCeVWW2gcVRg+58yaTUnizqbipZeX4uWhBEniBaoUX1Ioze52t7sRq6APio9V9MEaoWlVsFasRq0gltaAPuxms8lu0gcviE/FFOstVbSIxgcv6SU7EZqmdc7v9+9mJtNks51NTUH84ed889/PP+cmxP+d5FIbMJmNbpREu4WUkiTtCicKny0l1pIKmBzovF2S+hIJHX8iEu3hZJ5lNZGqyRrGSIQpq15AzF28jgpeY6yk6GVdrfFqdrD6Iw+QlB8g0YS2g7dyQmXM/IDhBhT0UCiRf59lfqmmDvzRt6kByV/m4JjtzuaujMUM2c5Z2d6JdKrRb3K2q6mA+oYVz8JnDdKPmmNthzkAk/lN63sYPgevrguc72aZX/L9C6x09GYyxBgCX4NlvyGUHOKELlm5rXeR1kchuChJt4SSwyddZRXgvwMGvYo4QSlk3/zkHD8UHxwVJA6zjZZqP8v8kK8OWLnIZtLyCAJagYC4rTGW/9Pqj92N/c+LUaAj27movwbi19tk/whRCIE7Q9vyI6yvRpftAKVTdUjOW40X3h5OXsKCdmFcx0xlLJoSuQngnrJe7Kcjm4OMq9FlC7CMmScQANuNvjfP3PjGXDBaUQmbp296S5L4DrpbrHN1T87ZVEZVCzg1FF0Ft+dKrlLukI+/c9ENo+TvlTDbYFvuKPtQ9+l052rXrgKoWkDAFnvh0wTOmYn8R5f4k/jN/fZiCM1tQx9jQQ4ANhqG4hiL0qIFTGViG9DKB7GYzgubnpofgYRwO+DFjh0Zin2m4b/97EDkXkc+f6xYAPX0KK2I/7fUQuwzuwo/L3AkcjugPNixC8cHf0FyPjWlItmLxWw4Ou9YsQCr5fijMGoD/zpdRy95HRysyXA74MWOnscpO4j2y3HAVisw85hX5+AFBRSHt4ShfLFkIMXTqyKFc46xdzQM6XbAi702a7sy04J0+feReMFKp5q9esYLCqAZYw/k14E/xcLLsFElaornTuJB0svMuJINy8xkIYuL+xPAlWRceH6+HX7THJ0djLUom46zREu7tTkxwmf/FdOZ/sh6Q8qvEAiHpm4PJ4a/doJe0gH1t+aHRgCzOvBvJedEK5OFE5jpm4AGP2a8Dxe3gGJ/pAutug9Gp6he92CsSsWBaEcxGx0FHytmIpuqGkOpldqNYQK8cSoXvd+xLxXADw0kf6UkJNFtdo5MOgaLjiQOQHcn+A6h5NuL2s0qsC2LOM75PcF3yr5STuBSAcGG+meA14K/CI21HcS4LBT6tv0QAh8Dr5l93AhZzG5ZJ4VxAqdZUEl9z7WJ4aN+svMvwHHL21UKTd1mqvChH7/Za5xzXBBKrUcB0TQ+Ulgkfbi/H/YT5EptrGzsEK7tR1B7ln9BBwckYfMiuSqklSznIuoIIOM42MQO+QnduCoFCI0bpkzjCjddHPN/F+2Yu+sd9bKNpVwHhbS3LluK/0zgfwD0xYI5dXuzlQAAAABJRU5ErkJggg==');
  }

  div.callout-warning.callout-style-default .callout-caption {
    background-color: #fcefdc
  }

  div.callout-tip {
    border-left-color: #02b875 !important;
  }

  div.callout-tip .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAADr0lEQVRYCe1XTWgTQRj9ZjZV8a9SPIkKgj8I1bMHsUWrqYLVg4Ue6v9BwZOxSYsIerFao7UiUryIqJcqgtpimhbBXoSCVxUFe9CTiogUrUp2Pt+3aUI2u5vdNh4dmMzOzHvvezuz8xNFM0mjnbXaNu1MvFWRXkXEyE6aYOYJpdW4IXuA4r0fo8qqSMDBU0v1HJUgVieAXxzCsdE/YJTdFcVIZQNMyhruOMJKXYFoLfIfIvVIMWdsrd+Rpd86ZmyzzjJmLStqRn0v8lzkb4rVIXvnpScOJuAn2ACC65FkPzEdEy4TPWRLJ2h7z4cArXzzaOdKlbOvKKX25Wl00jSnrwVxAg3o4dRxhO13RBSdNvH0xSARv3adTXbBdTf64IWO2vH0LT+cv4GR1DJt+DUItaQogeBX/chhbTBxEiZ6gftlDNXTrvT7co4ub5A6gp9HIcHvzTa46OS5fBeP87Qm0fQkr4FsYgVQ7Qg+ZayaDg9jhg1GkWj8RG6lkeSacrrHgDaxdoBiZPg+NXV/KifMuB6//JmYH4CntVEHy/keA6x4h4CU5oFy8GzrBS18cLJMXcljAKB6INjWsRcuZBWVaS3GDrqB7rdapVIeA+isQ57Eev9eCqzqOa81CY05VLd6SamW2wA2H3SiTbnbSxmzfp7WtKZkqy4mdyAlGx7ennghYf8voqp9cLSgKdqNfa6RdRsAAkPwRuJZNbpByn+RrJi1RXTwdi8RQF6ymDwGMAtZ6TVE+4uoKh+MYkcLsT0Hk8eAienbiGdjJHZTpmNjlbFJNKDVAp2fJlYju6IreQxQ08UJDNYdoLSl6AadO+fFuCQqVMB1NJwPm69T04Wv5WhfcWyfXQB+wXRs1pt+nCknRa0LVzSA/2B+a9+zQJadb7IyyV24YAxKp2Jqs3emZTuNnKxsah+uabKbMk7CbTgJx/zIgQYErIeTKRQ9yD9wxVof5YolPHqaWo7TD6tJlh7jQnK5z2n3+fGdggIOx2kaa2YI9QWarc5Ce1ipNWMKeSG4DysFF52KBmTNMmn5HqCFkwy34rDg05gDwgH3bBi+sgFhN/e8QvRn8kbamCOhgrZ9GJhFDgfcMHzFb6BAtjKpFhzTjwv1KCVuxHvCbsSiEz4CANnj84cwHdFXAbAOJ4LTSAawGWFn5tDhLMYz6nWeU2wJfIhmIJBefcd/A5FWQWGgrWzyORZ3Q6HuV+Jf0Bj+BTX69fm1zWgK7By1YTXchFDORywnfQ7GpzOo6S+qECrsx2ifVQAAAABJRU5ErkJggg==');
  }

  div.callout-tip.callout-style-default .callout-caption {
    background-color: #ccf1e3
  }

  div.callout-caution {
    border-left-color: #fd7e14 !important;
  }

  div.callout-caution .callout-icon::before {
    background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAAAXNSR0IArs4c6QAAAERlWElmTU0AKgAAAAgAAYdpAAQAAAABAAAAGgAAAAAAA6ABAAMAAAABAAEAAKACAAQAAAABAAAAIKADAAQAAAABAAAAIAAAAACshmLzAAACV0lEQVRYCdVWzWoUQRCuqp2ICBLJXgITZL1EfQDBW/bkzUMUD7klD+ATSHBEfAIfQO+iXsWDxJsHL96EHAwhgzlkg8nBg25XWb0zIb0zs9muYYWkoKeru+vn664fBqElyZNuyh167NXJ8Ut8McjbmEraKHkd7uAnAFku+VWdb3reSmRV8PKSLfZ0Gjn3a6Xlcq9YGb6tADjn+lUfTXtVmaZ1KwBIvFI11rRXlWlatwIAAv2asaa9mlB9wwygiDX26qaw1yYPzFXg2N1GgG0FMF8Oj+VIx7E/03lHx8UhvYyNZLN7BwSPgekXXLribw7w5/c8EF+DBK5idvDVYtEEwMeYefjjLAdEyQ3M9nfOkgnPTEkYU+sxMq0BxNR6jExrAI31H1rzvLEfRIdgcv1XEdj6QTQAS2wtstEALLG1yEZ3QhH6oDX7ExBSFEkFINXH98NTrme5IOaaA7kIfiu2L8A3qhH9zRbukdCqdsA98TdElyeMe5BI8Rs2xHRIsoTSSVFfCFCWGPn9XHb4cdobRIWABNf0add9jakDjQJpJ1bTXOJXnnRXHRf+dNL1ZV1MBRCXhMbaHqGI1JkKIL7+i8uffuP6wVQAzO7+qVEbF6NbS0LJureYcWXUUhH66nLR5rYmva+2tjRFtojkM2aD76HEGAD3tPtKM309FJg5j/K682ywcWJ3PASCcycH/22u+Bh7Aa0ehM2Fu4z0SAE81HF9RkB21c5bEn4Dzw+/qNOyXr3DCTQDMBOdhi4nAgiFDGCinIa2owCEChUwD8qzd03PG+qdW/4fDzjUMcE1ZpIAAAAASUVORK5CYII=');
  }

  div.callout-caution.callout-style-default .callout-caption {
    background-color: #ffe5d0
  }

  </style>
  <style type="text/css">
    .reveal div.sourceCode {
      margin: 0;
      overflow: auto;
    }
    .reveal div.hanging-indent {
      margin-left: 1em;
      text-indent: -1em;
    }
    .reveal .slide:not(.center) {
      height: 100%;
      overflow-y: auto;
    }
    .reveal .slide.scrollable {
      overflow-y: auto;
    }
    .reveal .footnotes {
      height: 100%;
      overflow-y: auto;
    }
    .reveal .slide .absolute {
      position: absolute;
      display: block;
    }
    .reveal .footnotes ol {
      counter-reset: ol;
      list-style-type: none; 
      margin-left: 0;
    }
    .reveal .footnotes ol li:before {
      counter-increment: ol;
      content: counter(ol) ". "; 
    }
    .reveal .footnotes ol li > p:first-child {
      display: inline-block;
    }
    .reveal .slide ul,
    .reveal .slide ol {
      margin-bottom: 0.5em;
    }
    .reveal .slide ul li,
    .reveal .slide ol li {
      margin-top: 0.4em;
      margin-bottom: 0.2em;
    }
    .reveal .slide ul[role="tablist"] li {
      margin-bottom: 0;
    }
    .reveal .slide ul li > *:first-child,
    .reveal .slide ol li > *:first-child {
      margin-block-start: 0;
    }
    .reveal .slide ul li > *:last-child,
    .reveal .slide ol li > *:last-child {
      margin-block-end: 0;
    }
    .reveal .slide .columns:nth-child(3) {
      margin-block-start: 0.8em;
    }
    .reveal blockquote {
      box-shadow: none;
    }
    .reveal .tippy-content>* {
      margin-top: 0.2em;
      margin-bottom: 0.7em;
    }
    .reveal .tippy-content>*:last-child {
      margin-bottom: 0.2em;
    }
    .reveal .slide > img.stretch.quarto-figure-center,
    .reveal .slide > img.r-stretch.quarto-figure-center {
      display: block;
      margin-left: auto;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-left,
    .reveal .slide > img.r-stretch.quarto-figure-left  {
      display: block;
      margin-left: 0;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-right,
    .reveal .slide > img.r-stretch.quarto-figure-right  {
      display: block;
      margin-left: auto;
      margin-right: 0; 
    }
  </style>
</head>
<body class="quarto-light">
  <div class="reveal">
    <div class="slides">

<section id="title-slide" class="quarto-title-block center">
  <h1 class="title">Good scientific practice: <a href="https://www.huber.embl.de/msmb/Chap-Design.html">Design of High Throughput Experiments and their Analysis</a></h1>

<div class="quarto-title-authors">
<div class="quarto-title-author">
<div class="quarto-title-author-name">
Susan Holmes, Wolfgang Huber 
</div>
</div>
</div>

  <p class="date">6/24/22</p>
</section>
<section id="design-of-high-throughput-experiments" class="slide level1">
<h1>Design of High Throughput Experiments</h1>
<div class="cell">

</div>
<table>
<colgroup>
<col style="width: 38%">
<col style="width: 61%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: center;">RA Fisher</th>
<th style="text-align: center;">Pioneer of Experimental Design</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: center;"><img data-src="fig/RAFisherSmoking.png" style="width:50.0%"></td>
<td style="text-align: center;"><em>To consult the statistician after an experiment is finished is often merely to ask him to conduct a post mortem examination. He can perhaps say what the experiment died of.</em></td>
</tr>
<tr class="even">
<td style="text-align: center;"><span class="citation" data-cites="fisher1935design">(<a href="#/further-reading" role="doc-biblioref" onclick="">Fisher 1935</a>)</span></td>
<td style="text-align: center;">(Presidential Address to the First Indian Statistical Congress, 1938. Sankhya 4, 14-17).</td>
</tr>
</tbody>
</table>
</section>
<section id="goals-for-this-lecture" class="slide level1">
<h1>Goals for this Lecture</h1>
<ul>
<li>Resource allocation and experimental design: an iterative process.</li>
<li>Dealing with the different types of variability; partitioning variability</li>
<li>Transformations</li>
<li>Types of experiments, studies, …</li>
<li>Power, sample size and efficiency.</li>
<li>Things to worry about: dependencies, batch effects, unwanted variation.</li>
<li>Compression, redundancy and sufficiency</li>
<li>Computational best practices</li>
</ul>
</section>
<section id="the-art-of-good-enough" class="slide level1">
<h1>The Art of “Good Enough”</h1>
<ul>
<li>Experimental design rationalizes the tradeoffs imposed by having finite resources.</li>
<li>Our measurement instruments have limited resolution and precision; often we don’t know these at the outset and have to collect preliminary data providing estimates.</li>
<li>Sample sizes are limited for practical, economic, and sometimes ethical reasons.</li>
<li>We may only be able to observe the phenomenon of interest indirectly rather than directly.</li>
<li>Our measurements may be overlaid with nuisance factors over which we have limited control.</li>
<li>There is little point in prescribing unrealistic ideals: we need to make pragmatic choices that are feasible.</li>
</ul>
</section>
<section id="types-of-studies-experiments" class="slide level1">
<h1>Types of studies / experiments</h1>
<h2 id="experiment">Experiment</h2>
<ul>
<li>everything is exquisitely controlled</li>
</ul>
<h2 id="retrospective-observational-studies">Retrospective observational studies</h2>
<ul>
<li>we take what we get, opportunistic; no control over study participants, assignment of important factors, confounding</li>
</ul>
<h2 id="prospective-controlled-studies">Prospective, controlled studies</h2>
<ul>
<li>e.g.&nbsp;clinical trials</li>
<li>randomization, blinding</li>
<li>ethical constraints (incl.&nbsp;money and time).</li>
</ul>
<h2 id="meta-analysis">Meta-analysis</h2>
<ul>
<li>We did not design the experiments or studies ourselves, nor collect the data.</li>
<li>Retrospective analysis of data that already happen to exist.</li>
</ul>
</section>
<section id="illustration-experiment" class="slide level1">
<h1>Illustration: experiment</h1>
<p>Well-characterized cell line growing in laboratory conditions on defined media, temperature and atmosphere.</p>
<p>We administer a precise amount of a drug, and after 72h we measure the activity of a specific pathway reporter.</p>
</section>
<section id="illustration-challenges-with-studies" class="slide level1">
<h1>Illustration: challenges with studies</h1>
<p>We recruited 200 patients that have a disease, fulfill inclusion criteria (e.g.&nbsp;age, comorbidities, mental capacity) and ask them to take a drug each day exactly at 6 am. After 3 months, we take an MRI scan and lots of other biomarkers to see whether and how the disease has changed or whether there were any other side effects.</p>
<ul>
<li>People may forget to take the pill or take it at the wrong time.</li>
<li>Some may feel that the disease got worse and stop taking the drug.</li>
<li>Some may feel that the disease got better and stop taking the drug.</li>
<li>Some may lead a healthy life-style, others eat junk food.</li>
<li>They have varying levels of disease to start with.</li>
<li>And all of these factors may be correlated with each other in unpredictable ways.</li>
</ul>
<p>What to do about this?</p>
</section>
<section id="examples" class="slide level1">
<h1>Examples</h1>
<ul>
<li>We modeled the sampling noise in RNA-seq and 16S rRNA with a Gamma-Poisson distribution.</li>
<li>We estimated sequencing depth bias with the library size factors.</li>
<li>We modeled sampling biases caused by the two different RNA-Seq protocols in the pasilla data (single-, paired-end) by introducing a blocking factor into our (generalized) linear model.</li>
</ul>
</section>
<section id="what-is-a-good-normalization-method" class="slide level1">
<h1>What is a good normalization method?</h1>
<div class="cell" data-hash="chap_design_cache/revealjs/norm_dcca801c956a4145ffa9e6d8d83876a8">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="fu">library</span>(<span class="st">"DESeq2"</span>)</span>
<span id="cb1-2"><a href="#cb1-2"></a><span class="fu">library</span>(<span class="st">"airway"</span>)</span>
<span id="cb1-3"><a href="#cb1-3"></a><span class="fu">library</span>(<span class="st">"ggplot2"</span>)</span>
<span id="cb1-4"><a href="#cb1-4"></a><span class="fu">library</span>(<span class="st">"dplyr"</span>)</span>
<span id="cb1-5"><a href="#cb1-5"></a><span class="fu">library</span>(<span class="st">"gridExtra"</span>)</span>
<span id="cb1-6"><a href="#cb1-6"></a><span class="fu">library</span>(<span class="st">"hexbin"</span>)</span>
<span id="cb1-7"><a href="#cb1-7"></a><span class="fu">data</span>(<span class="st">"airway"</span>)</span>
<span id="cb1-8"><a href="#cb1-8"></a>aw <span class="ot">=</span> <span class="fu">DESeqDataSet</span>(airway, <span class="at">design =</span> <span class="sc">~</span> cell <span class="sc">+</span> dex) <span class="sc">%&gt;%</span></span>
<span id="cb1-9"><a href="#cb1-9"></a>  estimateSizeFactors</span>
<span id="cb1-10"><a href="#cb1-10"></a><span class="fu">sizeFactors</span>(aw)</span>
<span id="cb1-11"><a href="#cb1-11"></a></span>
<span id="cb1-12"><a href="#cb1-12"></a>samples <span class="ot">=</span> <span class="fu">c</span>(<span class="st">"SRR1039513"</span>, <span class="st">"SRR1039517"</span>) </span>
<span id="cb1-13"><a href="#cb1-13"></a></span>
<span id="cb1-14"><a href="#cb1-14"></a>myScatterplot <span class="ot">=</span> <span class="cf">function</span>(x) {</span>
<span id="cb1-15"><a href="#cb1-15"></a>  <span class="fu">as_tibble</span>(x) <span class="sc">%&gt;%</span> </span>
<span id="cb1-16"><a href="#cb1-16"></a>  <span class="fu">mutate</span>(<span class="at">rs =</span> <span class="fu">rowSums</span>(x)) <span class="sc">%&gt;%</span></span>
<span id="cb1-17"><a href="#cb1-17"></a>  <span class="fu">filter</span>(rs <span class="sc">&gt;=</span> <span class="dv">2</span>) <span class="sc">%&gt;%</span></span>
<span id="cb1-18"><a href="#cb1-18"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">asinh</span>(SRR1039513), </span>
<span id="cb1-19"><a href="#cb1-19"></a>             <span class="at">y =</span> <span class="fu">asinh</span>(SRR1039517))) <span class="sc">+</span> <span class="fu">geom_hex</span>(<span class="at">bins =</span> <span class="dv">50</span>) <span class="sc">+</span></span>
<span id="cb1-20"><a href="#cb1-20"></a>    <span class="fu">coord_fixed</span>() <span class="sc">+</span> </span>
<span id="cb1-21"><a href="#cb1-21"></a>    <span class="fu">geom_abline</span>(<span class="at">slope =</span> <span class="dv">1</span>, <span class="at">intercept =</span> <span class="dv">0</span>, <span class="at">col =</span> <span class="st">"orange"</span>) <span class="sc">+</span> </span>
<span id="cb1-22"><a href="#cb1-22"></a>    <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb1-23"><a href="#cb1-23"></a>}</span>
<span id="cb1-24"><a href="#cb1-24"></a></span>
<span id="cb1-25"><a href="#cb1-25"></a><span class="fu">grid.arrange</span>(</span>
<span id="cb1-26"><a href="#cb1-26"></a>  <span class="fu">myScatterplot</span>(<span class="fu">counts</span>(aw)),</span>
<span id="cb1-27"><a href="#cb1-27"></a>  <span class="fu">myScatterplot</span>(<span class="fu">counts</span>(aw, <span class="at">normalized =</span> <span class="cn">TRUE</span>)),</span>
<span id="cb1-28"><a href="#cb1-28"></a>  <span class="at">ncol =</span> <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell fragment" data-hash="chap_design_cache/revealjs/norm_dcca801c956a4145ffa9e6d8d83876a8">
<div class="cell-output-display">
<p><img data-src="chap_design_files/figure-revealjs/norm-1.png" width="3000"></p>
</div>
</div>
<div class="cell" data-hash="chap_design_cache/revealjs/runDESeq_6f3f957e0197e9c0d62b66f4f941fe89">

</div>
<ul>
<li>If the normalization is ‘off’, we can have increased variability between replicates</li>
<li>… and/or apparent systematic differences between different conditions that are not real</li>
<li><span class="math inline">\(\to\)</span> false positives, false negatives</li>
</ul>
<p>What do we want from a good normalization method:</p>
<ul>
<li>remove technical variation</li>
<li>but <em>keep</em> biological variation</li>
</ul>
<p>Possible figure of merit?</p>
<ul>
<li>signal-to-noise ratio</li>
</ul>
</section>
<section id="occams-razor" class="slide level1">
<h1>Occam’s razor</h1>
<div class="columns">
<div class="column" style="width:40%;">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="fig/William_of_Ockham.png"></p>
<p></p><figcaption>William of Ockham</figcaption><p></p>
</figure>
</div>
</div><div class="column" style="width:&quot;60%;">
<p>If one can explain a phenomenon without assuming this or that hypothetical entity, there is no ground for assuming it.</p>
<p>One should always opt for an explanation in terms of the fewest possible causes, factors, or variables.</p>
</div>
</div>
</section>
<section id="error-models-noise-is-in-the-eye-of-the-beholder" class="slide level1">
<h1>Error models: Noise is in the eye of the beholder</h1>
<p>The efficiency of most biochemical or physical processes involving DNA-polymers depends on their sequence content, for instance, occurrences of long homopolymer stretches, palindromes, GC content.</p>
<p>These effects are not universal, but can also depend on <strong>factors</strong> like concentration, temperature, which enzyme is used, etc.</p>
<p>When looking at RNA-Seq data, should we treat GC content as noise or as bias?</p>
</section>
<section id="one-persons-noise-can-be-anothers-bias" class="slide level1">
<h1>One person’s noise can be another’s bias</h1>
<div class="columns">
<div class="column" style="width:10%;">
<div class="quarto-figure quarto-figure-right">
<figure>
<p><img data-src="fig/HeadsorTails.jpg" style="width:70.0%"></p>
</figure>
</div>
</div><div class="column" style="width:90%;">
<p>We may think that the outcome of tossing a coin is completely random.</p>
</div>
</div>
<p>If we meticulously registered the initial conditions of the coin flip and solved the mechanical equations, we could predict which side has a higher probability of coming up: noise becomes bias.</p>
<div style="text-align: center;">
<p><img data-src="fig/cointosser3.png" style="width:50.0%" alt="Tosser"> <a href="https://news.stanford.edu/pr/2004/diaconis-69.html">P. Diaconis, S. Holmes, et al.</a></p>
</div>
<p>We use probabilistic modelling as a method to bring some order into our ignorance – but let’s keep in mind that these models are a reflection of our subjective ignorance, not an objective property of the world.</p>
</section>
<section id="biological-versus-technical-replicates" class="slide level1">
<h1>Biological versus technical replicates</h1>
<p>Imagine we want to test whether a weight loss drug works. Which of the following designs is better?</p>
<ul>
<li><p>A person is weighed on milligram precision scales, with 20 replicates. He/she follows the diet, and four weeks later, (s)he is weighed again, with 20 replicates. The data are recorded by a specifically trained technician.</p></li>
<li><p>5 people weigh themselves on their bathroom scales and self-report the number. Four weeks later, they weigh themselves and report again. So we have 5 replicates.</p></li>
</ul>
<p>This example shows the difference between technological versus biological replicates.</p>
<p>Some design decisions in HT biology are similar, if more subtle:</p>
<ul>
<li>sequencing libraries</li>
<li>cell lines</li>
<li>CRISPR guides</li>
</ul>
</section>
<section id="quiz" class="slide level1">
<h1>Quiz</h1>
<p>For reliable variant calling with current sequencing technology, you need ca. <span class="math inline">\(30\times\)</span> coverage per genome.</p>
<p>In the <em>1000 genomes project</em>, the average depth of the data produced was 5.1 for 1,092 individuals. Why was that study design chosen?</p>
</section>
<section id="what-exactly-is-a-technical-versus-biological-replicate" class="slide level1">
<h1>What exactly is a technical versus biological replicate?</h1>
<p>In fact, the terminology is too simplistic. Error can creep in at many different levels:</p>
<ul>
<li>different labs,</li>
<li>different operators within one lab,</li>
<li>different technologies they use,</li>
<li>different machines from the same technology,</li>
<li>different variants of the protocol,</li>
<li>different strains, litters, sexes, individual animals</li>
</ul>
<p>We will use the notion of blocks.</p>
<p>If we know about these nuisance factors and have kept track of them, we can (should) include them explicitely as bias terms in our models.</p>
<p>If we did not keep track, we can try to use latent factor models (SVA, PEER, RUV) to identify them from the data.</p>
</section>
<section id="a-lack-of-units" class="slide level1">
<h1>A lack of units</h1>
<p>Pre-modern measurement systems measured lengths in feet, arms, inches (first joint of an index finger), weights in stones, volumes in multiples of the size of a wine jar, etc.</p>
<p>In the <a href="https://en.wikipedia.org/wiki/International_System_of_Units">International System of Units</a>: meters, seconds, kilograms, Ampères, … are defined based on universal physical constants. A meter measured by a lab in Australia using one instrument has the same meaning as a meter measured a year later by a lab in Stanford using a different instrument, by a space probe in the Kuiper belt, or ETs on Proxima Cen b.</p>
<p>Measurements in biology are, unfortunately, rarely that comparable.</p>
<p>Often, absolute values are not reported (these would require units), but only fold changes with regard to some local reference.</p>
<p>Even when absolute values exist (e.g., read counts in an RNA-Seq experiment) they usually do not translate into universal units such as molecules per cell or mole per milliliter.</p>
</section>
<section id="regular-and-catastrophic-noise" class="slide level1">
<h1>Regular and catastrophic noise</h1>
<p>Regular noise can be modelled by simple probability models such as independent normal distributions, Poissons, or mixtures such as Gamma-Poisson or Laplace.</p>
<p>We can use relatively straightforward methods to take noise into account in our data analyses.</p>
<p>The real world is more complicated: measurements can be completely off the scale (a sample swap, a contamination or a software bug), and multiple errors often come together: e.g., a whole microtiter plate went bad, affecting all data measured from it.</p>
<p>Such events are harder to model or even correct for – our best chance to deal with them is data quality assessment, outlier detection and documented removal.</p>
</section>
<section id="keeping-track-dailies" class="slide level1">
<h1>Keeping track: Dailies</h1>
<div class="columns">
<div class="column" style="width:30%;">
<div class="quarto-figure quarto-figure-right">
<figure>
<p><img data-src="fig/dailies_icon.png" style="width:50.0%"></p>
</figure>
</div>
</div><div class="column" style="width:70%;">
<p>A film director will view daily takes, to correct potential lighting, shooting issues before they affect too much footage. It is a good idea not to wait until all the runs of an experiment have been finished before looking at the data.</p>
</div>
</div>
<p>Intermediate data analyses and visualizations will track eventual unexpected sources of variation in the data and enable you to adjust the protocol.</p>
<p>It is important to be aware of sources of variation as they occur and adjust for them.</p>
<!--
# Phylochip example

<div style="text-align: center;"> 
![](/Users/whuber/svnco/CUBook/images/ThreeSets28s.png){width="50%"}
</div>
-->
</section>
<section id="rnai-screen-example" class="slide level1">
<h1>RNAi screen example</h1>
<div style="text-align: center;">
<p><img data-src="fig/RNAiScreen.png" style="width:50.0%"></p>
</div>
</section>
<section id="basic-principles-in-the-design-of-experiments" class="slide level1">
<h1>Basic principles in the design of experiments</h1>
<ul>
<li>Balancing</li>
<li>Avoidance of confounding</li>
<li>Blocking</li>
<li>Randomization</li>
<li>Attention to detail</li>
</ul>
</section>
<section id="clever-combinations-and-balancing-a-motivating-example" class="slide level1">
<h1>Clever combinations and balancing: a motivating example</h1>
<div class="columns">
<div class="column" style="width:20%;">
<div class="quarto-figure quarto-figure-right">
<figure>
<p><img data-src="fig/balancechem.png" style="width:50.0%"></p>
</figure>
</div>
</div><div class="column" style="width:80%;">
<p>A pharmacist’s balance weighing analogy (<span class="citation" data-cites="Hotelling1944">Hotelling (<a href="#/further-reading" role="doc-biblioref" onclick="">1944</a>)</span> and <span class="citation" data-cites="Mood1946">Mood (<a href="#/further-reading" role="doc-biblioref" onclick="">1946</a>)</span>).</p>
</div>
</div>
<p>Experimental design aims to maximize the available resources. One strategy is to capitalize on cancellations and symmetries. An interesting example is a weighing scheme devised by Hotelling.</p>
<p>Consider a set of eight objects of unknown weights <span class="math inline">\(m=(m_1,\ldots,m_8)\)</span>.</p>
<p>For our simulation study, we create a vector of true <span class="math inline">\(m\)</span> randomly.</p>
<div class="cell" data-hash="chap_design_cache/revealjs/HotellingsExpt_9195ffe776bb6114851c9612c9d59505">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a><span class="fu">library</span>(magrittr)</span>
<span id="cb2-2"><a href="#cb2-2"></a>m <span class="ot">=</span> <span class="fu">sample</span>(<span class="fu">seq</span>(<span class="dv">2</span>, <span class="dv">16</span>, <span class="at">by =</span> <span class="dv">2</span>), <span class="dv">8</span>)  <span class="sc">+</span> <span class="fu">round</span>(<span class="fu">rnorm</span>(<span class="dv">8</span>, <span class="dv">1</span>), <span class="dv">1</span>) <span class="sc">%T&gt;%</span> print</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.6 0.9 0.4 0.9 1.5 1.0 1.8 1.6</code></pre>
</div>
</div>
<h2 id="method-1-naive-method-eight-weighings">Method 1: Naive method, eight weighings</h2>
<p>First, let’s weigh each <span class="math inline">\(m_i\)</span> individually.</p>
<p>Suppose our scale has errors distributed normally with a SD of 0.1. We simulate the measurements (incl.&nbsp;individual errors) as follows. We also compute the root of the mean of their squares (RMS) as a measure of overall error.</p>
<div class="cell" data-hash="chap_design_cache/revealjs/SimpleWeighing_e0ccf22d4d22947b27b4db66ddd2f174">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a>X_1to8 <span class="ot">=</span> m <span class="sc">+</span> <span class="fu">rnorm</span>(<span class="fu">length</span>(m), <span class="at">mean =</span> <span class="dv">0</span>, <span class="at">sd =</span> <span class="fl">0.1</span>)</span>
<span id="cb4-2"><a href="#cb4-2"></a>X_1to8</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1]  8.359209 10.868363  4.298189 16.896361  3.454035 14.967027
[7]  7.857463 13.452812</code></pre>
</div>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1"></a>errors_1to8 <span class="ot">=</span> X_1to8 <span class="sc">-</span> m</span>
<span id="cb6-2"><a href="#cb6-2"></a>errors_1to8</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] -0.240791027 -0.031636891 -0.101811022 -0.003639091 -0.045965424
[6] -0.032972940  0.057463277 -0.147188095</code></pre>
</div>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1"></a>rms <span class="ot">=</span> <span class="cf">function</span>(x) <span class="fu">sqrt</span>(<span class="fu">mean</span>(x<span class="sc">^</span><span class="dv">2</span>))</span>
<span id="cb8-2"><a href="#cb8-2"></a><span class="fu">rms</span>(errors_1to8)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.1104119</code></pre>
</div>
</div>
<h2 id="method-2-hotellings-method-eight-weighings">Method 2: Hotelling’s method, eight weighings</h2>
<div class="cell" data-hash="chap_design_cache/revealjs/HotellingsMethod_77db965d6847200f16b104810b34e9eb">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1"></a><span class="fu">library</span>(<span class="st">"survey"</span>)</span>
<span id="cb10-2"><a href="#cb10-2"></a>h8 <span class="ot">=</span> <span class="fu">hadamard</span>(<span class="dv">6</span>)</span>
<span id="cb10-3"><a href="#cb10-3"></a>coef8 <span class="ot">=</span> <span class="dv">2</span> <span class="sc">*</span> h8 <span class="sc">-</span> <span class="dv">1</span></span>
<span id="cb10-4"><a href="#cb10-4"></a>coef8</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>     [,1] [,2] [,3] [,4] [,5] [,6] [,7] [,8]
[1,]    1    1    1    1    1    1    1    1
[2,]    1   -1    1   -1    1   -1    1   -1
[3,]    1    1   -1   -1    1    1   -1   -1
[4,]    1   -1   -1    1    1   -1   -1    1
[5,]    1    1    1    1   -1   -1   -1   -1
[6,]    1   -1    1   -1   -1    1   -1    1
[7,]    1    1   -1   -1   -1   -1    1    1
[8,]    1   -1   -1    1   -1    1    1   -1</code></pre>
</div>
</div>
<p>We use the columns of this matrix to guide our new weighing scheme.</p>
<p>The first column says we should put all 8 pieces on one side of the balance and weigh them, call this <code>Y[1]</code>.</p>
<p>The second column says: place 1, 3, 5, 7 on one side of the balance and on the other and evaluate the difference. Call the result <code>Y[2]</code>.</p>
<p>And so on.</p>
<p>Each of the eight weighings has a normal error with sd = 0.1.</p>
<div class="cell" data-hash="chap_design_cache/revealjs/unnamed-chunk-6_c01a6e03fb24b976ecb01ffd7bbe7416">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1"></a>Y <span class="ot">=</span> m  <span class="sc">%*%</span> coef8 <span class="sc">+</span> <span class="fu">rnorm</span>(<span class="fu">length</span>(m), <span class="at">mean =</span> <span class="dv">0</span>, <span class="at">sd =</span> <span class="fl">0.1</span>)</span>
<span id="cb12-2"><a href="#cb12-2"></a>mhat <span class="ot">=</span> Y <span class="sc">%*%</span> <span class="fu">t</span>(coef8) <span class="sc">/</span> <span class="dv">8</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Now, because we know the true <span class="math inline">\(m\)</span>, we can compute the errors and their sum of squares:</p>
<div class="cell" data-hash="chap_design_cache/revealjs/Hoterrors_5790340669cdde747bbb4c816b1affba">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1"></a>errors_hotel <span class="ot">=</span> <span class="fu">as.vector</span>(mhat) <span class="sc">-</span> m</span>
<span id="cb13-2"><a href="#cb13-2"></a>errors_hotel</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1]  0.052211813  0.035408972 -0.051291112  0.027410736 -0.028415118
[6] -0.006958278  0.034178628 -0.049007538</code></pre>
</div>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1"></a><span class="fu">rms</span>(errors_hotel)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 0.03838579</code></pre>
</div>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1"></a><span class="fu">rms</span>(errors_1to8) <span class="sc">/</span> <span class="fu">rms</span>(errors_hotel)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 2.876375</code></pre>
</div>
</div>
</section>
<section id="proper-simulation" class="slide level1">
<h1>Proper simulation</h1>
<p>We saw that the second method had RMS about 3 times smaller than the first. Were we just lucky?</p>
<ul>
<li><p>Repeat Hotelling’s weighing experiment B = 10,000 times with different <span class="math inline">\(m\)</span> and look at the sampling distributions of the variances of the errors in both schemes</p></li>
<li><p>Guess at the relationship between the RMS results.</p></li>
</ul>
<div class="cell columns column-output-location" data-hash="chap_design_cache/revealjs/bootstrapHotelling_36542db4bd8009160f8d7787043e2f7e">
<div class="column">
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode numberSource r code-overflow-wrap number-lines code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1"></a>tcoef8 <span class="ot">=</span> <span class="fu">t</span>(coef8) <span class="sc">/</span> <span class="dv">8</span></span>
<span id="cb19-2"><a href="#cb19-2"></a>errs <span class="ot">=</span> <span class="fu">replicate</span>(<span class="dv">10000</span>, {</span>
<span id="cb19-3"><a href="#cb19-3"></a>  m <span class="ot">=</span> <span class="fu">sample</span>(<span class="fu">seq</span>(<span class="dv">2</span>, <span class="dv">16</span>, <span class="at">by =</span> <span class="dv">2</span>), <span class="dv">8</span>) <span class="sc">+</span> <span class="fu">round</span>(<span class="fu">rnorm</span>(<span class="dv">8</span>, <span class="dv">1</span>), <span class="dv">1</span>)</span>
<span id="cb19-4"><a href="#cb19-4"></a>  X_1to8 <span class="ot">=</span> m <span class="sc">+</span> <span class="fu">rnorm</span>(<span class="fu">length</span>(m), <span class="at">mean =</span> <span class="dv">0</span>, <span class="at">sd =</span> <span class="fl">0.1</span>)</span>
<span id="cb19-5"><a href="#cb19-5"></a>  err_1to8 <span class="ot">=</span> X_1to8 <span class="sc">-</span> m</span>
<span id="cb19-6"><a href="#cb19-6"></a>  Y <span class="ot">=</span> coef8 <span class="sc">%*%</span> m <span class="sc">+</span> <span class="fu">rnorm</span>(<span class="fu">length</span>(m), <span class="at">mean =</span> <span class="dv">0</span>, <span class="at">sd =</span> <span class="fl">0.1</span>)</span>
<span id="cb19-7"><a href="#cb19-7"></a>  mhat <span class="ot">=</span> tcoef8 <span class="sc">%*%</span> Y</span>
<span id="cb19-8"><a href="#cb19-8"></a>  err_hotel <span class="ot">=</span> mhat <span class="sc">-</span> m</span>
<span id="cb19-9"><a href="#cb19-9"></a>  <span class="fu">c</span>(<span class="fu">rms</span>(err_1to8), <span class="fu">rms</span>(err_hotel))  </span>
<span id="cb19-10"><a href="#cb19-10"></a>})</span>
<span id="cb19-11"><a href="#cb19-11"></a></span>
<span id="cb19-12"><a href="#cb19-12"></a><span class="fu">library</span>(<span class="st">"ggplot2"</span>)</span>
<span id="cb19-13"><a href="#cb19-13"></a><span class="fu">library</span>(<span class="st">"tibble"</span>)</span>
<span id="cb19-14"><a href="#cb19-14"></a><span class="fu">ggplot</span>(<span class="fu">tibble</span>(<span class="at">ratio =</span> errs[<span class="dv">1</span>,] <span class="sc">/</span> errs[<span class="dv">2</span>,]), </span>
<span id="cb19-15"><a href="#cb19-15"></a>       <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">log2</span>(ratio))) <span class="sc">+</span></span>
<span id="cb19-16"><a href="#cb19-16"></a>  <span class="fu">geom_histogram</span>(<span class="at">bins =</span>  <span class="dv">50</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div><div class="column">
<div class="cell-output-display">
<p><img data-src="chap_design_files/figure-revealjs/bootstrapHotelling-1.png" style="width:100.0%"></p>
</div>
</div>
</div>
<p>We say that the second scheme is more efficient than the first by a factor of <span class="math inline">\(\sqrt{8}\)</span> because the RMS errors generated by the measurement are <span class="math inline">\(\sqrt{8}\)</span> times lower (<span class="math inline">\(\log_2\sqrt{8}\approx\)</span> 1.5).</p>
</section>
<section id="one-factor-at-a-time" class="slide level1">
<h1>One factor at a time?</h1>
<div class="columns">
<div class="column" style="width:33%;">
<p>Ibn Sina (Avicenna) <img data-src="fig/Avicenna.png" style="width:100.0%" alt="Avicenna"></p>
</div><div class="column" style="width:33%;">
<p>Physician Scientist <img data-src="fig/medicus.jpeg" style="height:75.0%" alt="The Physician"></p>
</div><div class="column" style="width:30%;">
<p>His <em>Canon of Medicine (1020)</em> lists seven rules of experimental design, including the need for controls and replication, the danger of confounding and the necessity of changing only one factor at a time.</p>
</div>
</div>
<h2 id="this-dogma-was-overthrown-in-the-20th-century-by-ra-fisher.">This dogma was overthrown in the 20th century by RA Fisher.</h2>
</section>
<section id="comparing-two-levels-of-one-factor-healthy-or-diseased." class="slide level1">
<h1>Comparing two levels of one factor: healthy or diseased.</h1>
<div class="cell" data-hash="chap_design_cache/revealjs/illustrateboxplots_ba9e9e70f4894427c21ad4ff02918100">

</div>
<div class="cell" data-hash="chap_design_cache/revealjs/p0p0eff_9ccdfe52e90b4525a8911f946f04d3b3">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1"></a><span class="fu">grid.arrange</span>(p0, p0effect, <span class="at">ncol =</span> <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img data-src="chap_design_files/figure-revealjs/p0p0eff-1.png" style="width:100.0%"></p>
</div>
</div>
<p>However, suppose the experiment was done in two “batches”, and we color the data according that:</p>
<div class="cell" data-hash="chap_design_cache/revealjs/p0p0batch_da9eb4f6d5c166bc8a2d5e10c5962c87">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1"></a><span class="fu">grid.arrange</span>(p0, p0batch, <span class="at">ncol =</span> <span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img data-src="chap_design_files/figure-revealjs/p0p0batch-1.png" style="width:100.0%"></p>
</div>
</div>
<p>We cannot conclude because we are in the presence of <strong>confounding</strong>.</p>
<p>Now suppose the experiment has higher noise levels, and the same number of points as in the previous study. Then, the sample sizes (2 x 6) is not enough. With the same error and a bigger sample size (2 x 60), it looks better.</p>
<div class="cell" data-hash="chap_design_cache/revealjs/unnamed-chunk-9_e30a4e237877e3e32f82c0c5e6ad339f">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1"></a><span class="fu">grid.arrange</span>(p1, pN1, <span class="at">ncol=</span><span class="dv">2</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<p><img data-src="chap_design_files/figure-revealjs/unnamed-chunk-9-1.png" style="width:100.0%"></p>
</div>
</div>
<p>The experiment at <span class="math inline">\(n_1=n_2=6\)</span> is not <strong>powerful</strong> enough.</p>
<div class="cell" data-hash="chap_design_cache/revealjs/unnamed-chunk-10_85f1da68549ae23b7584bb2288329102">
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1"></a><span class="fu">t.test</span>(<span class="st">`</span><span class="at">Biomarker B</span><span class="st">`</span> <span class="sc">~</span> states, <span class="at">data =</span> batdef)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
    Welch Two Sample t-test

data:  Biomarker B by states
t = -2.7115, df = 9.9937, p-value = 0.02189
alternative hypothesis: true difference in means between group healthy and group tumor is not equal to 0
95 percent confidence interval:
 -1.1592407 -0.1133796
sample estimates:
mean in group healthy   mean in group tumor 
             1.549076              2.185387 </code></pre>
</div>
</div>
<p>With the same effect size and a larger sample size, we have the <strong>power</strong> to see the difference:</p>
<div class="cell" data-hash="chap_design_cache/revealjs/unnamed-chunk-11_d37b5a9d29d34bf4861422dc9796946a">
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1"></a><span class="fu">t.test</span>(<span class="st">`</span><span class="at">Biomarker C</span><span class="st">`</span><span class="sc">~</span> states_60, <span class="at">data =</span> dfN)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
    Welch Two Sample t-test

data:  Biomarker C by states_60
t = -7.822, df = 116.83, p-value = 2.56e-12
alternative hypothesis: true difference in means between group healthy and group tumor is not equal to 0
95 percent confidence interval:
 -0.5516679 -0.3287520
sample estimates:
mean in group healthy   mean in group tumor 
             1.493643              1.933853 </code></pre>
</div>
</div>
</section>
<section id="lessons-from-these-toy-examples" class="slide level1">
<h1>Lessons from these toy examples</h1>
<ul>
<li>Success of an experiment: seeing difference if it is truely there.</li>
<li>Knowing where the difference comes from.<br>
</li>
<li>True positive rate: power.</li>
<li>Spread (e.g., variances) matters as much as the locations (e.g., means).</li>
</ul>
<p>Depends on:</p>
<ol type="1">
<li>Effect size (<strong>unchangeable</strong>)</li>
<li>Control and documentation of all factors (block effects, date / time / operator etc.).</li>
<li>Noise (variability in measurements)</li>
<li>Sample size: remember the standard error of the mean is <span class="math display">\[\frac{\sigma^2}{n}\]</span></li>
</ol>
</section>
<section id="decomposition-of-variability-analysis-of-variance." class="slide level1">
<h1>Decomposition of variability: analysis of variance.</h1>

<img data-src="chap_design_files/figure-revealjs/block-1.png" style="width:100.0%" class="r-stretch"></section>
<section id="blocking-the-case-of-paired-experiments." class="slide level1">
<h1>Blocking: the case of paired experiments.</h1>
<div class="columns">
<div class="column" style="width:40%;">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="fig/maizeDarwin.png" style="width:100.0%"></p>
<p></p><figcaption>ZeaMays</figcaption><p></p>
</figure>
</div>
</div><div class="column" style="width:55%;">
<p>Each pot in Darwin’s <em>Zea Mays</em> experiment is a block, only the factor of interest should be different (pollination method), all other factors should be kept equal within a block.<br> A balanced design is an experimental design where all the different factor combinations have the same number of observation replicates. Such data are particularly easy to analyse because the effect of each factor is identifiable.<br> When there are (likely) nuisance factors, it is good to make sure they are balanced with the factors of interest. Sometimes this is inconvenient or impractical for logistic or economic reasons – but in such cases analysts are on thin ice and need to proceed with caution.|</p>
</div>
</div>
</section>
<section id="comparing-paired-versus-unpaired-design" class="slide level1">
<h1>Comparing paired versus unpaired design</h1>
<p>When comparing various possible designs, we do power simulations.</p>
<p>Here, we suppose the sample size is 15 in each group and the <span>effect size</span> is 0.2. We also need to make assumptions about the standard deviations of the measurements, here we suppose both groups have the same sd=0.25.</p>
<div class="cell" data-hash="chap_design_cache/revealjs/zeamays_787455509a0536d92c37c4cb3aabe85c">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1"></a>n <span class="ot">=</span> <span class="dv">15</span></span>
<span id="cb27-2"><a href="#cb27-2"></a>effect <span class="ot">=</span> <span class="fl">0.2</span></span>
<span id="cb27-3"><a href="#cb27-3"></a>pots   <span class="ot">=</span> <span class="fu">rnorm</span>(<span class="dv">15</span>, <span class="dv">0</span>, <span class="dv">1</span>)</span>
<span id="cb27-4"><a href="#cb27-4"></a>noiseh <span class="ot">=</span> <span class="fu">rnorm</span>(<span class="dv">15</span>, <span class="dv">0</span>, <span class="fl">0.25</span>)</span>
<span id="cb27-5"><a href="#cb27-5"></a>noisea <span class="ot">=</span> <span class="fu">rnorm</span>(<span class="dv">15</span>, <span class="dv">0</span>, <span class="fl">0.25</span>)</span>
<span id="cb27-6"><a href="#cb27-6"></a>autoz  <span class="ot">=</span> pots <span class="sc">+</span> noisea</span>
<span id="cb27-7"><a href="#cb27-7"></a>hybrid <span class="ot">=</span> pots <span class="sc">+</span> effect <span class="sc">+</span> noiseh</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Perform a simple and a paired <span class="math inline">\(t\)</span>-test. Which is more powerful in this case?</p>
<div class="cell" data-hash="chap_design_cache/revealjs/unnamed-chunk-13_aad5068be457ba5c65b2aa4949763169">
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1"></a><span class="fu">t.test</span>(hybrid, autoz, <span class="at">paired =</span> <span class="cn">FALSE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
    Welch Two Sample t-test

data:  hybrid and autoz
t = 0.58292, df = 27.908, p-value = 0.5646
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -0.6118805  1.0985542
sample estimates:
 mean of x  mean of y 
 0.2016872 -0.0416497 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1"></a><span class="fu">t.test</span>(hybrid, autoz, <span class="at">paired =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>
    Paired t-test

data:  hybrid and autoz
t = 3.3113, df = 14, p-value = 0.005145
alternative hypothesis: true mean difference is not equal to 0
95 percent confidence interval:
 0.08572565 0.40094807
sample estimates:
mean difference 
      0.2433369 </code></pre>
</div>
</div>
<p>Maybe we were just lucky (due to randomness in our simulation) here.</p>
<p>To confirm, run a parametric bootstrap experiment. Generate data as above <code>B=1000</code> times and compute the average probability of rejection, if allowing for a false positive rate <span class="math inline">\(\alpha=0.05\)</span>.</p>
<div class="cell" data-hash="chap_design_cache/revealjs/bootstrapPower_937540a97f9f5632ff86f70be0e50fdb">
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1"></a>pvals <span class="ot">=</span> <span class="fu">replicate</span>(<span class="dv">1000</span>, {</span>
<span id="cb32-2"><a href="#cb32-2"></a>  pots   <span class="ot">=</span> <span class="fu">rnorm</span>(<span class="dv">15</span>,<span class="dv">0</span>,<span class="dv">1</span>)</span>
<span id="cb32-3"><a href="#cb32-3"></a>  noiseh <span class="ot">=</span> <span class="fu">rnorm</span>(<span class="dv">15</span>,<span class="dv">0</span>,<span class="fl">0.25</span>)</span>
<span id="cb32-4"><a href="#cb32-4"></a>  noisea <span class="ot">=</span> <span class="fu">rnorm</span>(<span class="dv">15</span>,<span class="dv">0</span>,<span class="fl">0.25</span>)</span>
<span id="cb32-5"><a href="#cb32-5"></a>  autoz  <span class="ot">=</span> pots <span class="sc">+</span> noisea</span>
<span id="cb32-6"><a href="#cb32-6"></a>  hybrid <span class="ot">=</span> pots <span class="sc">+</span> effect <span class="sc">+</span> noiseh</span>
<span id="cb32-7"><a href="#cb32-7"></a>  <span class="fu">sapply</span>(<span class="fu">c</span>(<span class="cn">FALSE</span>, <span class="cn">TRUE</span>), <span class="cf">function</span>(prd) </span>
<span id="cb32-8"><a href="#cb32-8"></a>    <span class="fu">setNames</span>(<span class="fu">t.test</span>(hybrid, autoz, <span class="at">paired =</span> prd)<span class="sc">$</span>p.value, <span class="fu">paste</span>(prd))</span>
<span id="cb32-9"><a href="#cb32-9"></a>  )</span>
<span id="cb32-10"><a href="#cb32-10"></a>}) </span>
<span id="cb32-11"><a href="#cb32-11"></a><span class="fu">rowMeans</span>(pvals <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>FALSE  TRUE 
0.000 0.538 </code></pre>
</div>
</div>
<p>We can plot the p-values obtained using both methods:</p>
<div class="cell" data-hash="chap_design_cache/revealjs/pvaluescompare_05f21d5fa7c437b20f14d15d68a6b7d8">
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1"></a><span class="fu">library</span>(<span class="st">"tidyr"</span>)</span>
<span id="cb34-2"><a href="#cb34-2"></a><span class="fu">pivot_longer</span>(<span class="fu">as_tibble</span>(<span class="fu">t</span>(pvals)), <span class="at">cols =</span> <span class="fu">rownames</span>(pvals)) <span class="sc">%&gt;%</span></span>
<span id="cb34-3"><a href="#cb34-3"></a>  <span class="fu">transform</span>(<span class="at">name =</span> <span class="fu">paste</span>(<span class="st">"paired ="</span>, name)) <span class="sc">%&gt;%</span></span>
<span id="cb34-4"><a href="#cb34-4"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> value, <span class="at">fill =</span> name)) <span class="sc">+</span> </span>
<span id="cb34-5"><a href="#cb34-5"></a>    <span class="fu">geom_histogram</span>(<span class="at">binwidth =</span> <span class="fl">0.02</span>, <span class="at">alpha =</span> <span class="fu">exp</span>(<span class="sc">-</span><span class="dv">1</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell fragment" data-hash="chap_design_cache/revealjs/pvaluescompare_05f21d5fa7c437b20f14d15d68a6b7d8">
<div class="cell-output-display">
<p><img data-src="chap_design_files/figure-revealjs/pvaluescompare-1.png" style="width:80.0%"></p>
</div>
</div>
</section>
<section id="exercises" class="slide level1">
<h1>Exercises</h1>
<ul>
<li>Make a function that compares the power of the two types of tests for different values of the effect size, sample size, standard deviation among the pots.</li>
<li>Use your function to find out which of the standard deviations (pots or noise) has the largest effect on the improvement produced by pairing for <span class="math inline">\(n = 15\)</span>.</li>
</ul>
<div class="cell" data-hash="chap_design_cache/revealjs/powerPairedUnpaired_6fce41755c8294cb625914c0dda33928">
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1"></a>powercomparison <span class="ot">=</span> <span class="cf">function</span>(<span class="at">potsd =</span> <span class="dv">1</span>, <span class="at">noisesd =</span> <span class="fl">0.25</span>, <span class="at">n =</span> <span class="dv">15</span>, <span class="at">effect =</span> <span class="fl">0.2</span>,</span>
<span id="cb35-2"><a href="#cb35-2"></a>                           <span class="at">B =</span> <span class="dv">1000</span>, <span class="at">pthresh =</span> <span class="fl">0.05</span>) {</span>
<span id="cb35-3"><a href="#cb35-3"></a>  pv <span class="ot">=</span> <span class="fu">replicate</span>(B, {</span>
<span id="cb35-4"><a href="#cb35-4"></a>    pots   <span class="ot">=</span> <span class="fu">rnorm</span>(n, <span class="dv">0</span>, potsd)</span>
<span id="cb35-5"><a href="#cb35-5"></a>    noiseh <span class="ot">=</span> <span class="fu">rnorm</span>(n, <span class="dv">0</span>, noisesd)</span>
<span id="cb35-6"><a href="#cb35-6"></a>    noisea <span class="ot">=</span> <span class="fu">rnorm</span>(n, <span class="dv">0</span>, noisesd)</span>
<span id="cb35-7"><a href="#cb35-7"></a>    autoz  <span class="ot">=</span> pots <span class="sc">+</span> noisea</span>
<span id="cb35-8"><a href="#cb35-8"></a>    hybrid <span class="ot">=</span> pots <span class="sc">+</span> effect <span class="sc">+</span> noiseh</span>
<span id="cb35-9"><a href="#cb35-9"></a>    <span class="fu">sapply</span>(<span class="fu">c</span>(<span class="cn">FALSE</span>, <span class="cn">TRUE</span>), <span class="cf">function</span>(prd) </span>
<span id="cb35-10"><a href="#cb35-10"></a>      <span class="fu">setNames</span>(<span class="fu">t.test</span>(hybrid, autoz, <span class="at">paired =</span> prd)<span class="sc">$</span>p.value, <span class="fu">paste</span>(prd))</span>
<span id="cb35-11"><a href="#cb35-11"></a>    )</span>
<span id="cb35-12"><a href="#cb35-12"></a>  })</span>
<span id="cb35-13"><a href="#cb35-13"></a>  <span class="fu">rowMeans</span>(pv <span class="sc">&lt;</span> pthresh)</span>
<span id="cb35-14"><a href="#cb35-14"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Here are a few examples showing that when the pots sd is smaller than the noise sd, pairing hardly makes a difference. If the pots variability is larger than that of the measurement noise, pairing makes a big difference.</p>
<div class="cell" data-hash="chap_design_cache/revealjs/unnamed-chunk-16_caf25afa6247ca48b86e1d7110ef3c0f">
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1"></a><span class="fu">powercomparison</span>(<span class="at">potsd =</span> <span class="fl">0.5</span>,  <span class="at">noisesd =</span> <span class="fl">0.25</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>FALSE  TRUE 
0.037 0.515 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1"></a><span class="fu">powercomparison</span>(<span class="at">potsd =</span> <span class="fl">0.25</span>, <span class="at">noisesd =</span> <span class="fl">0.5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>FALSE  TRUE 
0.134 0.175 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1"></a><span class="fu">powercomparison</span>(<span class="at">potsd =</span> <span class="fl">0.25</span>, <span class="at">noisesd =</span> <span class="fl">0.1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>FALSE  TRUE 
0.511 0.998 </code></pre>
</div>
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1"></a><span class="fu">powercomparison</span>(<span class="at">potsd =</span> <span class="fl">0.1</span>,  <span class="at">noisesd =</span> <span class="fl">0.25</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>FALSE  TRUE 
0.472 0.510 </code></pre>
</div>
</div>
<p>For 100 plants of each type and the two SDs being 0.5, the power of the paired test is nearly 80%.</p>
<div class="cell" data-hash="chap_design_cache/revealjs/unnamed-chunk-17_61e9c988ea13b13366da7a83e353d417">
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1"></a><span class="fu">powercomparison</span>(<span class="at">potsd =</span> <span class="fl">0.5</span>, <span class="at">noisesd =</span> <span class="fl">0.5</span>, <span class="at">n =</span> <span class="dv">100</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>FALSE  TRUE 
0.512 0.787 </code></pre>
</div>
</div>
<p>Lesson: take into account a natural pairing of the observations – for instance, studies of patients before and after a treatment.</p>
<p>What can be done when pairing is not available?</p>
<p>Try to create pairs of subjects that have as much similarity as possible through matching age, gender (sex), background health etc. One is treated, the other serves as a control.</p>
</section>
<section id="block-what-you-can-randomize-what-you-cannot" class="slide level1">
<h1>“Block what you can, randomize what you cannot”</h1>
<p>(George Box, 1978)</p>
<p>Often we don’t know which nuisance factors will be important, or we cannot plan for them ahead of time.</p>
<p>In such cases, randomization is a practical strategy: at least in the limit of large enough sample size, the effect of any nuisance factor should average out.</p>
<h2 id="randomized-complete-block-design">Randomized Complete Block Design</h2>
<p><img data-src="fig/CompleteRB3.png" style="width:50.0%"></p>
<p>“The design space is divided into uniform units to account for any variation so that observed differences within units are largely due to true differences between treatments. Treatments are then assigned at random to the subjects in the blocks - once in each block. The defining feature of the Randomized Complete Block Design is that each block sees each treatment exactly once.” (<a href="https://pbgworks.org/sites/pbgworks.org/files/RandomizedCompleteBlockDesignTutorial.pdf">Trudi Grant</a>)</p>
<h2 id="balanced-incomplete-block-design">Balanced Incomplete Block Design</h2>
<p><img data-src="fig/balincomblock.png" style="width:50.0%"></p>
<h2 id="complete-factorial-latin-squares">Complete Factorial Latin Squares</h2>
<div class="quarto-figure quarto-figure-left">
<figure>
<p><img data-src="fig/LatinSquare9.png" style="width:50.0%"></p>
<p></p><figcaption>sudoku anyone?</figcaption><p></p>
</figure>
</div>
<p>(Latin square: n × n array filled with n different symbols, each occurring exactly once in each row and exactly once in each column)</p>
</section>
<section id="randomization-decreases-bias" class="slide level1">
<h1>Randomization decreases bias</h1>
<ul>
<li>Humans are bad at assigning treatments truly at random</li>
<li>Random assignment reduces unconscious bias (special samples treated differently, “balancing things out”, …)</li>
<li>Randomization also helps with unknown nuisance factors.</li>
</ul>
<h2 id="randomization-helps-inference">Randomization helps inference</h2>
<ul>
<li>if the sample is randomly generated from a population, we can infer something about the population we drew from</li>
</ul>
<h2 id="random-does-not-mean-haphazardly">Random does not mean haphazardly</h2>
<ul>
<li>Need to use a random number generator and a seed.</li>
</ul>
</section>
<section id="controls-positive-and-negative-why" class="slide level1">
<h1>Controls, positive and negative: why?</h1>
<p>We often need to remove variation due to unknown factors, or decompose variability according to its different sources; this is classically done using analysis of variance that can accomodatefactors such as subject or batch effects.</p>
<p>Usually these decompositions require at least 3 replicate measurements in each <code>cell</code>.</p>
<p>Removal of effects from unknown sources can only be done through the use of <strong>negative controls</strong>.</p>
<p>Calibration of the effect size in an experiment also requires the use of <strong>positive controls</strong>; spike-ins (for instance External RNA Control Consortium controls as used in <span class="citation" data-cites="Risso2014">Risso et al. (<a href="#/further-reading" role="doc-biblioref" onclick="">2014</a>)</span>) where a known quantity or a known expression level aid in these calibrations and are a standard part of many experimental protocols.</p>
<p>Validation in independent data / independent lines of argument</p>
<p>If it is too good to be true, it usually is….</p>
<p><a href="https://pubpeer.com/publications/E6CD4F8EE374B7D2D398A542684C78">Synonymous mutations in representative yeast genes are mostly strongly non-neutral</a></p>
</section>
<section id="how-many-replicates-do-i-need" class="slide level1">
<h1>How many replicates do I need?</h1>
<p>We saw how we can use preliminary data and simulations to calculate how many replicates are needed to achieve, say, a 80% true positive rate, if we know the effect size.</p>
<p>Now, recall the discussion of experiments versus studies.</p>
<p>For the cell line experiment, we might get the correct result already from one replicate; usually we’ll do two or three to be sure.</p>
<p>On the other hand, for the study, our intuition tells us that there is so much uncontrolled variability that 20 is likely far too few, and we may need 200 or 2,000 patients for a reliable result. The number depends on the amount of uncontrolled variability, and the effect size. A pragmatic approach is to check out previous successful (or unsuccessful) experiments or studies that did something comparable and use simulations, subsampling or bootstrapping to get an estimate of our proposed new study’s power. Here are more details about how to go about this in practice.</p>
</section>
<section id="power-depends-on-sample-sizes-effect-sizes-and-variability." class="slide level1">
<h1>Power depends on sample sizes, effect sizes and variability.</h1>
<ul>
<li>Example of two batches with different variability.</li>
<li>Examples of one treatment where the variance is larger.</li>
<li>Need to accomodate <strong>heteroscedasticity</strong> (unequal variances) with different sample sizes.</li>
</ul>
</section>
<section id="effective-sample-size-for-dependent-data." class="slide level1">
<h1>Effective sample size for dependent data.</h1>
<p>Dependent data require <strong>larger</strong> samples (intuition: each sample contains <strong>less</strong> information)</p>
<p>A sample of independent observations is more informative than the same number of dependent observations.</p>
<p>Suppose you want to do an opinion poll by knocking at people’s doors and asking them a question.</p>
<ul>
<li>Scenario 1: pick <span class="math inline">\(n\)</span> people at <span class="math inline">\(n\)</span> random places throughout the country.</li>
<li>Scenario 2: to save travel time, you pick <span class="math inline">\(n/3\)</span> random places and then at each of these interview three people who live next door to each other.</li>
</ul>
<p>In both cases, the number of people polled is <span class="math inline">\(n\)</span>, but if we assume that people living in the same neighborhood are more likely to have the same opinion, the data from Scenario 2 are (positively) correlated. To explore this, let’s do a simulation.</p>
<div class="cell" data-hash="chap_design_cache/revealjs/effective_sample_size_sim_8537e040792d6402f8d8f10f4f24aec7">
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1"></a>doPoll <span class="ot">=</span> <span class="cf">function</span>(<span class="at">n =</span> <span class="dv">100</span>, <span class="at">numPeoplePolled =</span> <span class="dv">12</span>) {</span>
<span id="cb46-2"><a href="#cb46-2"></a>  opinion <span class="ot">=</span> <span class="fu">sort</span>(<span class="fu">rnorm</span>(n))</span>
<span id="cb46-3"><a href="#cb46-3"></a>  i1 <span class="ot">=</span> <span class="fu">sample</span>(n, numPeoplePolled)</span>
<span id="cb46-4"><a href="#cb46-4"></a>  i2 <span class="ot">=</span> <span class="fu">sample</span>(<span class="fu">seq</span>(<span class="dv">3</span>, n, <span class="at">by =</span> <span class="dv">3</span>), numPeoplePolled <span class="sc">/</span> <span class="dv">3</span>)</span>
<span id="cb46-5"><a href="#cb46-5"></a>  i2 <span class="ot">=</span> <span class="fu">c</span>(i2, i2 <span class="sc">-</span> <span class="dv">1</span>, i2 <span class="sc">-</span> <span class="dv">2</span>)</span>
<span id="cb46-6"><a href="#cb46-6"></a>  <span class="fu">c</span>(<span class="at">independent =</span> <span class="fu">mean</span>(opinion[i1]), <span class="at">correlated =</span> <span class="fu">mean</span>(opinion[i2]))</span>
<span id="cb46-7"><a href="#cb46-7"></a>}</span>
<span id="cb46-8"><a href="#cb46-8"></a>responses <span class="ot">=</span> <span class="fu">replicate</span>(<span class="dv">5000</span>, <span class="fu">doPoll</span>())</span>
<span id="cb46-9"><a href="#cb46-9"></a><span class="fu">ggplot</span>(<span class="fu">pivot_longer</span>(<span class="fu">as_tibble</span>(<span class="fu">t</span>(responses)), <span class="at">cols =</span> <span class="fu">rownames</span>(responses)), </span>
<span id="cb46-10"><a href="#cb46-10"></a>       <span class="fu">aes</span>(<span class="at">x =</span> value, <span class="at">col =</span> name)) <span class="sc">+</span> <span class="fu">geom_density</span>() <span class="sc">+</span></span>
<span id="cb46-11"><a href="#cb46-11"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>) <span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">"Opinion poll result"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="chap_design_files/figure-revealjs/effective_sample_size_sim-1.png" style="width:80.0%"></p>
<p></p><figcaption>Density estimates for the polling result using the two sampling methods. The correlated method has higher spread. The truth is indicated by the vertical line.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>There are 100 people in the country, of which in the first approach (<code>i1</code>) we randomly sample 12. In the second approach, we sample 4 people as well as two neighbors for each (<code>i2</code>). The <em>opinion</em> in our case is a real number, normally distributed in the population with mean 0 and standard deviation 1.</p>
<p>We model the spatio-sociological structure of our country by sorting the houses from most negative to most positive opinion in the first line of the <code>doPoll</code> function.</p>
<h2 id="time-course-equalize-the-dependence-structure-by-taking-more-frequent-samples-when-there-is-more-going-on.">Time course: equalize the dependence structure by taking more frequent samples when there is more “going on”.</h2>
<div class="quarto-figure quarto-figure-left">
<figure>
<p><img data-src="fig/SamplingDesignSmall.png" style="width:60.0%"></p>
<p></p><figcaption>Response to a transient perturbation</figcaption><p></p>
</figure>
</div>
</section>
<section id="mean-variance-relationships-and-transformations" class="slide level1">
<h1>Mean-Variance Relationships and Transformations</h1>
<p>You’ve seen examples for data transformations that compress or stretch the space of quantitativemeasurements in such a way that the measurements’ noise variance (<strong>not:</strong> the signal variance) is more similar throughout – i.e., no longer highly dependent on the mean value.</p>
<p>The mean-variance relationship of a measurement technology can in principle be any function, but in many cases, the following prototyic relationships hold:</p>
<ol type="1">
<li><p>constant: the variance is independent of the mean (“additive noise”, <span class="math inline">\(Y=\mu+\varepsilon\)</span>)</p></li>
<li><p>Poisson: the variance is proportional to to the mean (“shot noise”, <span class="math inline">\(N\sim\text{Poi}(\mu)\)</span>)</p></li>
<li><p>quadratic: the standard deviation is proportional to the mean, therefore the variance grows quadratically (“multiplicative noise”, <span class="math inline">\(Y=\mu \cdot e^\varepsilon\)</span>)</p></li>
</ol>
<p>Do you know examples for biological assays (or measurement technologies) whose data show these types of relationships?</p>
<p>Real data can also be affected by a combination of these. For instance:</p>
<ul>
<li><p>with DNA microarrays, the fluorescence intensities are subject to a combination of background noise that is largely independent of the signal (additive noise), and multiplicative noise whose standard deviation is proportional to the signal <span class="citation" data-cites="RockeDurbin:2001">(<a href="#/further-reading" role="doc-biblioref" onclick="">Rocke and Durbin 2001</a>)</span>. Therefore, for bright spots the multiplicative noise dominates, whereas for faint ones, the background.</p></li>
<li><p>in RNA-Seq, low counts are dominated by Poisson noise, high counts by the multiplicative noise.</p></li>
</ul>
<p>We recall from Chapter 5 that for data with a mean-variance relationship <span class="math inline">\(v(\mu)\)</span> the variance-stabilizing transformation <span class="math inline">\(g:\mathbb{R}\to\mathbb{R}\)</span> fulfills the condition <span class="math display">\[g'(x)=\frac{1}{\sqrt{v(x)}}\]</span></p>
<ol type="a">
<li>What are the variance-stabilizing transformations associated with the above three prototypic mean-variance relationships?</li>
<li>What is the variance stabilizing transformation that is appropriate for <span class="math inline">\(v(\mu) = v_0 + c\,\mu^2\)</span>, where <span class="math inline">\(v_0&gt;0\)</span> is a positive constant?</li>
</ol>
</section>
<section id="data-quality-assessment-and-quality-control" class="slide level1">
<h1>Data quality assessment and quality control</h1>
<ul>
<li>data quality assessment (QA): steps taken to measure and monitor data quality</li>
<li>data quality control (QC): removing bad data</li>
</ul>
<p>These pervade all phases of an analysis, from assembling the raw data over transformation, summarization, model fitting, hypothesis testing or screening for ‘’hits’’ to interpretation. QA-related questions include:</p>
<ul>
<li>How do the marginal distributions of the variables look (histograms, ECDF plots)?</li>
<li>How do their joint distributions look (scatter plots, pairs plot)?</li>
<li>How well do replicates agree (as compared to different biological conditions)? Are the magnitudes of the differences between several conditions plausible?</li>
<li>Is there evidence of batch effects? These could be of a categorical (stepwise) or continuous (gradual) nature, e.g., due to changes in experimental reagents, protocols or environmental factors. Factors associated with such effects may be explicitly known, or unkown and latent; often they are somewhere in between (e.g., when a measurement apparatus slowly degrades over time)</li>
</ul>
<p>For the last two sets of questions, heatmaps, principal component plots and other ordination plots are useful.</p>
<div class="columns">
<div class="column" style="width:50%;">
<div class="quarto-figure quarto-figure-left">
<figure>
<p><img data-src="fig/1896_Ford_Quadricycle.jpeg"></p>
</figure>
</div>
</div><div class="column" style="width:50%;">
<p>Henry Ford’s (possibly apocryphal) quote: ‘’If I had asked people what they wanted, they would have said faster horses.’’ (https://corporate.ford.com/history.html) expresses the view of quality as <strong>fitness for purpose</strong>, versus <strong>adherence to specifications</strong>.</p>
<p>It is not easy to nail down (mathematically define) quality, and the word is used with many meanings. See also http://en.wikipedia.org/wiki/Quality_(business)</p>
</div>
</div>
</section>
<section id="longitudinal-data" class="slide level1">
<h1>Longitudinal Data</h1>
<p>Longitudinal data have time as a covariate.</p>
<ul>
<li><p>handful of time points (e.g.&nbsp;response of a cell line measured 48h, 72h and 84h after exposure to a drug) vs a long and densely sampled time series (e.g.&nbsp;patch clamp data in electrophysiology or movie from life cell microscopy).</p></li>
<li><p>in 1st case, think of time as just another experimental factor, like concentration or choice of drug. One analysis strategy could be to first identify the “best”, or biologically most indicative, time point, and then focus on that. Or whether there is any effect at all, regardless of the time.</p></li>
<li><p>in 2nd case, time series, we’ll often want to fit dynamical models to the data. We have many choices:</p>
<ul>
<li>(Hidden) Markov models</li>
<li>Change point detection</li>
<li>Ordinary differential equations or reaction-diffusion models</li>
<li>Piece-wise deterministic stochastic processes</li>
<li>Autoregressive models</li>
<li>Non-parametric smoothing followed by clustering or classification into prototypic shapes</li>
</ul></li>
</ul>
</section>
<section id="dont-pretend-you-are-dumb" class="slide level1">
<h1>Don’t Pretend You Are Dumb</h1>
<p>There is some attraction to seemingly “unbiased” approaches that analyse the data at hand without any reference to what is already known. Such tendencies are reinforced by the fact that statistical methods have often been developed to be generic, for instance, working of a general matrix without specific reference to what the rows and column might signify.</p>
<p>Generic approaches are a good way to get started, and for analyses that are highly powered and straightforward, such an an approach might work out. But often, it is wasteful. Recall the example of an RNA-Seq experiment for differential expression. As we saw in the <em>RNA-Seq</em> and <em>Hypothesis Testing</em> lectures, we could perform a hypothesis test for differential expression for each annotated gene, regardless of its read counts, and then run a multiple testing method that treats all tests as exchangeable. But this is inefficient - we can improve our detection power by filtering out, or downweighting, the hypotheses with lower test power, or with lower prior probability of being alternative.</p>
<p>Other examples include:</p>
<ul>
<li><p>Penalisation or feature selection in high-dimensional regression or classification. It is easy to use schemes that treat all features the same, for instance, standardize all of them to zero mean and unit variance. But sometime we know that some classes of features are likely to be more or less informative than others.</p></li>
<li><p>You can also use graphs to represent “other” data (networks) and use approaches like the group or graph lasso to structure your penalties in high-dimensional modelling.</p></li>
<li><p>Unsupervised clustering of your objects of interests (samples, genes, etc.) and subsequent search for over-represented annotations. Instead, just check whether the members of groups defined by the annotations are more similar than two randomly picked objects.</p></li>
</ul>
</section>
<section id="leaky-pipelines-and-statistical-sufficiency" class="slide level1">
<h1>Leaky pipelines and statistical sufficiency</h1>
<div class="columns">
<div class="column" style="width:40%;">
<p><img data-src="fig/leakypipeline.png" style="width:70.0%"></p>
</div><div class="column" style="width:55%;">
<p>Data analysis pipelines in high-throughput biology often work as <strong>funnels</strong> that successively summarise and compress the data. In high-throughput sequencing, we may start with individual sequencing reads, then align them to a reference, then only count the aligned reads for each position, summarise positions to genes (or other kinds of regions), then “normalize” these numbers by library size, etc.</p>
</div>
</div>
<p>At each step, we loose some information, and it is important to make sure we still have enough information for the task at hand. The problem is particularly burning if we use a data pipeline built from a series of separate components without enough care being taken ensuring that all the information necessary for ulterior steps is conserved.</p>
<p>Statisticians have a concept for whether certain summaries enable the reconstruction of all the relevant information in the data: <strong>sufficiency</strong>. E.g., in a Bernoulli random experiment with a known number of trials, <span class="math inline">\(n\)</span>, the number of successes is a sufficient statistic for estimating the probability of success <span class="math inline">\(p\)</span>.</p>
<p>In a 4-state Markov chain (A,C,G,T) such as the one we saw in Lecture 2, what are the sufficient statistics for the estimation of the transition probabilities?</p>
<p>Iterative approaches akin to what we saw when we used the EM algorithm can sometimes help avoid information loss. For instance, when analyzing mass spectroscopy data, a first run guesses at peaks individually for every sample. After this preliminary spectra-spotting, another iteration allows us to borrow strength from the other samples to spot spectra that may have been labeled as noise.</p>
</section>
<section id="sharpen-your-tools-reproducible-research" class="slide level1">
<h1>Sharpen Your Tools: Reproducible Research</h1>
<p>Analysis projects often begin with a simple script, perhaps to try out a few initial ideas or explore the quality of the pilot data. Then more ideas are added, more data come in, other datasets are integrated, more people become involved. Eventually the paper needs to be written, figures be redone properly, and the analysis be saved for the scientific record and to document its integrity. Here are a few tools that can help with such a process.</p>
<p><strong>Use an integrated development environment.</strong> <span>RStudio</span> is a great choice; there are also other platforms such as Emacs or Eclipse.</p>
<p><strong>Use literate programming</strong> tools such as <span>Rmarkdown</span> or Jupyter. This is more readable (for yourself and for others) than burying explanations and usage instructions in comments in the source code or in separate README files, in addition you can directly embed figures and tables in these documents. Such documents are often good starting points for the supplementary material of your paper.</p>
<p><strong>Anticipate re-engineering of the data formats and the software.</strong> The first version of how you choose to represent the data and structure the analysis workflows will rarely be the best. Don’t be afraid to make a clean cut and redesign them as soon as you notice that you are doing a lot of awkward data manipulations or repetitive steps. This is time well-invested. Sometimes it also helps to unearth bugs.</p>
<p><strong>Reuse existing tools.</strong> Don’t reinvent the wheel and rather spend your time on things that are actually new. Before implementing a ‘heuristic’ or a temporary hack that analyses your data in a certain way, spend a couple of minutes researching to see if something like this hasn’t been done before. More often than not, it has, and there is a clean, scalable, and already tested solution.</p>
<p><strong>Use version control</strong>, such as git. This takes some time to learn, but in the long run will be infinitely better than all your self-grown attempts at managing evolving code with version numbers, switches and the like. Moreover, this is also the sanest option for collaborative work on code, and it provides an extra backup of your codebase, especially if the server is distinct from your workplace machine.</p>
<p><strong>Use functions</strong> rather than copy-pasting stretches of code.</p>
<p><strong>Use the R package system.</strong> Soon you’ll note recurring function or variable definitions that you want to share between your individual scripts. It is fine to use the R function to manage them initially, but it is never to early to move them into your own package at the latest when you find yourself starting to write README files or long emails explaining others how to use some script or another. Assembling existing code into an R package is not hard by any means, and offers you many goodies including standardized and convenient ways to provide documentation, to show code usage examples, to test the correct functioning of your code, and to version it. Quite likely you’ll soon appreciate the benefit of using namespaces.</p>
<p><strong>Centralize the location of the raw data files and streamline the derivation of intermediate data.</strong> Store the input data at a centralized file server that is professionally backed up. Mark the files as read-only. Have a clear and linear workflow for computing the derived data (e.g., normalised, summarised, transformed etc.) from the raw files, and store these in a separate directory. Anticipate that this workflow will need to be re-run several times, and version it. Use or similar tools to mirror these files on your personal computer.</p>
<p><strong>Integration.</strong> When developing downstream analysis ideas that bring together several different data types, you don’t want to do the conversion from data type specific formats to the representations that machine learning or generic statistical methods use each time on an ad hoc basis. Have a <em>recipe</em> script that assembles the different ingredients and cooks them up as an easily consumable matrix, data frame or Bioconductor.</p>
<p><strong>Keep a hyperlinked webpage with an index of all analyses.</strong> This is helpful for collaborators (especially if the page and the analysis can be accessed via a web browser) and also a good starting point for the methods part of your paper. Structure it in chronological or logical order, or a combination of both.</p>
</section>
<section id="data-representation" class="slide level1">
<h1>Data representation</h1>
<p>Combining all the data so it is ready for analysis or visualisation often involves a lot of shuffling around of the data, until they are in the right shape and format for an analytical algorithm or a graphics routine.</p>
<p>Errors can occur, lost labels, lost information: be safe, redundancy is good.</p>
</section>
<section id="wide-vs-long-table-format" class="slide level1">
<h1>Wide vs long table format</h1>
<p>Recall Hiiragi data (for space reasons we print only the first five rows and columns):</p>
<div class="cell" data-hash="chap_design_cache/revealjs/wide_f344a97f1febd4828bdccf5d71255301">
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1"></a><span class="fu">data</span>(<span class="st">"x"</span>, <span class="at">package =</span> <span class="st">"Hiiragi2013"</span>)</span>
<span id="cb47-2"><a href="#cb47-2"></a>dfw <span class="ot">=</span> <span class="fu">as.data.frame</span>(<span class="fu">exprs</span>(x))</span>
<span id="cb47-3"><a href="#cb47-3"></a>dfw[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>               1 E3.25   2 E3.25  3 E3.25   4 E3.25   5 E3.25
1415670_at    4.910459  7.526672 6.956328  6.424048  5.105808
1415671_at    9.768979  9.144228 9.295010 11.059831  9.376749
1415672_at   10.411893 10.918942 9.495738 10.317996 11.143684
1415673_at    5.618108  6.439416 6.730465  4.914527  5.619778
1415674_a_at  7.541891  8.380285 8.480580  7.977363  8.650312</code></pre>
</div>
<div class="sourceCode cell-code" id="cb49"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1"></a><span class="fu">dim</span>(dfw)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 45101   101</code></pre>
</div>
</div>
<p>This dataframe has columns of data, one for each sample (annotated by the column names). Its rows correspond to the probes, annotated by the probe identifiers. This is an example for a data table in <em>wide format</em>.</p>
<div class="cell" data-hash="chap_design_cache/revealjs/pivot_longer_675672c3963ad637dfdef0398540b162">
<div class="sourceCode cell-code" id="cb51"><pre class="sourceCode numberSource r number-lines code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1"></a>dfw<span class="sc">$</span>probeid <span class="ot">=</span> <span class="fu">rownames</span>(dfw)</span>
<span id="cb51-2"><a href="#cb51-2"></a>dfl <span class="ot">=</span> <span class="fu">pivot_longer</span>(dfw, </span>
<span id="cb51-3"><a href="#cb51-3"></a>            <span class="at">cols =</span> <span class="fu">colnames</span>(x),</span>
<span id="cb51-4"><a href="#cb51-4"></a>            <span class="at">names_to =</span> <span class="st">"Embryonic Day"</span>) <span class="sc">%T&gt;%</span> print</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 4,555,201 × 3
   probeid    `Embryonic Day` value
   &lt;chr&gt;      &lt;chr&gt;           &lt;dbl&gt;
 1 1415670_at 1 E3.25          4.91
 2 1415670_at 2 E3.25          7.53
 3 1415670_at 3 E3.25          6.96
 4 1415670_at 4 E3.25          6.42
 5 1415670_at 5 E3.25          5.11
 6 1415670_at 6 E3.25          5.86
 7 1415670_at 7 E3.25          5.06
 8 1415670_at 8 E3.25          4.57
 9 1415670_at 9 E3.25          8.12
10 1415670_at 10 E3.25         5.46
# … with 4,555,191 more rows</code></pre>
</div>
</div>
<p>In <code>dfl</code>, each row corresponds to exactly one measured value, stored in the column named <code>value</code>. Then there are additional columns that store the associated covariates.</p>
<p>Now suppose we want to store somewhere not only the probe identifiers but also the associated gene symbols.</p>
<p>In <code>dfw</code>, we could stick them as an additional column, and perhaps also throw in the genes’ ENSEMBL identifier for good measure. But now we have a problem: the dataframe now has some columns that represent different samples, and others that refer to information for all samples (the gene symbol and identifier) and we somehow have to know this when interpreting the dataframe. This is what Hadley Wickham calls <strong>untidy data</strong>.</p>
<p>In contrast, in <code>dfl</code>, we can add these columns, yet still know that each row forms exactly one observation, and all information associated with that observation is in the same row.</p>
</section>
<section id="tidy-data" class="slide level1">
<h1>Tidy data</h1>
<p>In tidy data</p>
<ol type="1">
<li><p>each variable forms a column,</p></li>
<li><p>each observation forms a row,</p></li>
<li><p>each type of observational unit forms a table.</p></li>
</ol>
</section>
<section id="using-tidy-data-wisely" class="slide level1">
<h1>Using tidy data wisely</h1>
<p>See also https://www.huber.embl.de/msmb/Chap-Design.html#tidy-data-using-it-wisely</p>
<h2 id="efficiency-lack-of">Efficiency (lack of)</h2>
<p>Even though there are only 45101 probe-gene symbol relationships, we are storing them 4555201 times in the rows of <code>dfl</code>.</p>
<h2 id="standardization-lack-of">Standardization (lack of)</h2>
<p>We may choose to call these columns <code>probeid</code> and <code>geneid</code>, but the next person might call them <code>probe_id</code> and <code>gene_id</code>, or even something completely different. When we find a dataframe that was made by someone else and that contains such-named columns, we may hope, but have no guarantees, that these are valid gene symbols.</p>
<p>Addressing such issues is behind the object-oriented design of the specialized data structures in Bioconductor, such as the class <code>SummarizedExperiment</code>.</p>
<h2 id="matrices-versus-dataframes">Matrices versus dataframes</h2>
<p>Representation of the matrix may also simply be more natural and mathematically stringent - e.g., if we want to apply PCA / dimension reduction, or classification.</p>
</section>
<section id="summary" class="slide level1">
<h1>Summary</h1>
<p>There are two types of variation in an experiment or study: those of interest, those that are unwanted.</p>
<p>We usually cannot rid ourselves of all the unwanted variation, but we saw how we can used balanced randomized designs, data transformations.</p>
<p>Reproducible workflows</p>
<p>Open science: make data and code available</p>
</section>
<section id="further-reading" class="slide level1 smaller scrollable">
<h1>Further Reading</h1>
<p>Book chapter</p>
<p>Lecture and chapter presented merely a pragmatic introduction to design, there are many book long treatments of the subject that offer detailed advice on setting up experiments: <span class="citation" data-cites="wu2011experiments">Wu and Hamada (<a href="#/further-reading" role="doc-biblioref" onclick="">2011</a>)</span>; <span class="citation" data-cites="box1978statistics">Box, Hunter, and Hunter (<a href="#/further-reading" role="doc-biblioref" onclick="">1978</a>)</span>; <span class="citation" data-cites="glass2007experimental">Glass (<a href="#/further-reading" role="doc-biblioref" onclick="">2007</a>)</span></p>
<h2 id="references">References</h2>
<div class="footer footer-default">

</div>
<div id="refs" class="references csl-bib-body hanging-indent" role="doc-bibliography">
<div id="ref-box1978statistics" class="csl-entry" role="doc-biblioentry">
Box, George EP, William G Hunter, and J Stuart Hunter. 1978. <em>Statistics for Experimenters: An Introduction to Design, Data Analysis, and Model Building</em>. John Wiley &amp; Sons.
</div>
<div id="ref-fisher1935design" class="csl-entry" role="doc-biblioentry">
Fisher, Ronald Aylmer. 1935. <em>The Design of Experiments</em>. Oliver &amp; Boyd.
</div>
<div id="ref-glass2007experimental" class="csl-entry" role="doc-biblioentry">
Glass, David J. 2007. <em>Experimental Design for Biologists</em>. Cold Spring Harbor Laboratory Press.
</div>
<div id="ref-Hotelling1944" class="csl-entry" role="doc-biblioentry">
Hotelling, Harold. 1944. <span>“Some Improvements in Weighing and Other Experimental Techniques.”</span> <em>The Annals of Mathematical Statistics</em> 15 (3): 297–306.
</div>
<div id="ref-Mood1946" class="csl-entry" role="doc-biblioentry">
Mood, Alexander M. 1946. <span>“On Hotelling’s Weighing Problem.”</span> <em>The Annals of Mathematical Statistics</em>, 432–46.
</div>
<div id="ref-Risso2014" class="csl-entry" role="doc-biblioentry">
Risso, Davide, John Ngai, Terence P Speed, and Sandrine Dudoit. 2014. <span>“Normalization of RNA-Seq Data Using Factor Analysis of Control Genes or Samples.”</span> <em>Nature Biotechnology</em> 32 (9): 896–902.
</div>
<div id="ref-RockeDurbin:2001" class="csl-entry" role="doc-biblioentry">
Rocke, David M, and Blythe Durbin. 2001. <span>“A Model for Measurement Error for Gene Expression Arrays.”</span> <em>Journal of Computational Biology</em> 8 (6): 557–69.
</div>
<div id="ref-wu2011experiments" class="csl-entry" role="doc-biblioentry">
Wu, CF Jeff, and Michael S Hamada. 2011. <em>Experiments: Planning, Analysis, and Optimization</em>. Vol. 552. John Wiley &amp; Sons.
</div>
</div>
</section>
    </div>
  </div>

  <script>window.backupDefine = window.define; window.define = undefined;</script>
  <script src="chap_design_files/libs/revealjs/dist/reveal.js"></script>
  <!-- reveal.js plugins -->
  <script src="chap_design_files/libs/revealjs/plugin/quarto-line-highlight/line-highlight.js"></script>
  <script src="chap_design_files/libs/revealjs/plugin/pdf-export/pdfexport.js"></script>
  <script src="chap_design_files/libs/revealjs/plugin/reveal-menu/menu.js"></script>
  <script src="chap_design_files/libs/revealjs/plugin/reveal-menu/quarto-menu.js"></script>
  <script src="chap_design_files/libs/revealjs/plugin/quarto-support/support.js"></script>
  

  <script src="chap_design_files/libs/revealjs/plugin/notes/notes.js"></script>
  <script src="chap_design_files/libs/revealjs/plugin/search/search.js"></script>
  <script src="chap_design_files/libs/revealjs/plugin/zoom/zoom.js"></script>
  <script src="chap_design_files/libs/revealjs/plugin/math/math.js"></script>
  <script>window.define = window.backupDefine; window.backupDefine = undefined;</script>

  <script>

      // Full list of configuration options available at:
      // https://revealjs.com/config/
      Reveal.initialize({
'controlsAuto': true,
'previewLinksAuto': false,
'smaller': true,
'pdfSeparateFragments': false,
'autoAnimateEasing': "ease",
'autoAnimateDuration': 1,
'autoAnimateUnmatched': true,
'menu': {"side":"left","useTextContentForMissingTitles":true,"markers":false,"loadIcons":false,"custom":[{"title":"Tools","icon":"<i class=\"fas fa-gear\"></i>","content":"<ul class=\"slide-menu-items\">\n<li class=\"slide-tool-item active\" data-item=\"0\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.fullscreen(event)\"><kbd>f</kbd> Fullscreen</a></li>\n<li class=\"slide-tool-item\" data-item=\"1\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.speakerMode(event)\"><kbd>s</kbd> Speaker View</a></li>\n<li class=\"slide-tool-item\" data-item=\"2\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.overview(event)\"><kbd>o</kbd> Slide Overview</a></li>\n<li class=\"slide-tool-item\" data-item=\"3\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.overview(event)\"><kbd>e</kbd> PDF Export Mode</a></li>\n<li class=\"slide-tool-item\" data-item=\"4\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.keyboardHelp(event)\"><kbd>?</kbd> Keyboard Help</a></li>\n</ul>"}],"openButton":true},
'smaller': true,
 
        // Display controls in the bottom right corner
        controls: false,

        // Help the user learn the controls by providing hints, for example by
        // bouncing the down arrow when they first encounter a vertical slide
        controlsTutorial: false,

        // Determines where controls appear, "edges" or "bottom-right"
        controlsLayout: 'edges',

        // Visibility rule for backwards navigation arrows; "faded", "hidden"
        // or "visible"
        controlsBackArrows: 'faded',

        // Display a presentation progress bar
        progress: true,

        // Display the page number of the current slide
        slideNumber: false,

        // 'all', 'print', or 'speaker'
        showSlideNumber: 'all',

        // Add the current slide number to the URL hash so that reloading the
        // page/copying the URL will return you to the same slide
        hash: true,

        // Start with 1 for the hash rather than 0
        hashOneBasedIndex: false,

        // Flags if we should monitor the hash and change slides accordingly
        respondToHashChanges: true,

        // Push each slide change to the browser history
        history: true,

        // Enable keyboard shortcuts for navigation
        keyboard: true,

        // Enable the slide overview mode
        overview: true,

        // Disables the default reveal.js slide layout (scaling and centering)
        // so that you can use custom CSS layout
        disableLayout: false,

        // Vertical centering of slides
        center: false,

        // Enables touch navigation on devices with touch input
        touch: true,

        // Loop the presentation
        loop: false,

        // Change the presentation direction to be RTL
        rtl: false,

        // see https://revealjs.com/vertical-slides/#navigation-mode
        navigationMode: 'linear',

        // Randomizes the order of slides each time the presentation loads
        shuffle: false,

        // Turns fragments on and off globally
        fragments: true,

        // Flags whether to include the current fragment in the URL,
        // so that reloading brings you to the same fragment position
        fragmentInURL: false,

        // Flags if the presentation is running in an embedded mode,
        // i.e. contained within a limited portion of the screen
        embedded: false,

        // Flags if we should show a help overlay when the questionmark
        // key is pressed
        help: true,

        // Flags if it should be possible to pause the presentation (blackout)
        pause: true,

        // Flags if speaker notes should be visible to all viewers
        showNotes: false,

        // Global override for autoplaying embedded media (null/true/false)
        autoPlayMedia: null,

        // Global override for preloading lazy-loaded iframes (null/true/false)
        preloadIframes: null,

        // Number of milliseconds between automatically proceeding to the
        // next slide, disabled when set to 0, this value can be overwritten
        // by using a data-autoslide attribute on your slides
        autoSlide: 0,

        // Stop auto-sliding after user input
        autoSlideStoppable: true,

        // Use this method for navigation when auto-sliding
        autoSlideMethod: null,

        // Specify the average time in seconds that you think you will spend
        // presenting each slide. This is used to show a pacing timer in the
        // speaker view
        defaultTiming: null,

        // Enable slide navigation via mouse wheel
        mouseWheel: false,

        // The display mode that will be used to show slides
        display: 'block',

        // Hide cursor if inactive
        hideInactiveCursor: true,

        // Time before the cursor is hidden (in ms)
        hideCursorTime: 5000,

        // Opens links in an iframe preview overlay
        previewLinks: false,

        // Transition style (none/fade/slide/convex/concave/zoom)
        transition: 'none',

        // Transition speed (default/fast/slow)
        transitionSpeed: 'default',

        // Transition style for full page slide backgrounds
        // (none/fade/slide/convex/concave/zoom)
        backgroundTransition: 'none',

        // Number of slides away from the current that are visible
        viewDistance: 3,

        // Number of slides away from the current that are visible on mobile
        // devices. It is advisable to set this to a lower number than
        // viewDistance in order to save resources.
        mobileViewDistance: 2,

        // The "normal" size of the presentation, aspect ratio will be preserved
        // when the presentation is scaled to fit different resolutions. Can be
        // specified using percentage units.
        width: 1050,

        height: 700,

        // Factor of the display size that should remain empty around the content
        margin: 0.1,

        math: {
          mathjax: 'https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.0/MathJax.js',
          config: 'TeX-AMS_HTML-full',
          tex2jax: {
            inlineMath: [['\\(','\\)']],
            displayMath: [['\\[','\\]']],
            balanceBraces: true,
            processEscapes: false,
            processRefs: true,
            processEnvironments: true,
            preview: 'TeX',
            skipTags: ['script','noscript','style','textarea','pre','code'],
            ignoreClass: 'tex2jax_ignore',
            processClass: 'tex2jax_process'
          },
        },

        // reveal.js plugins
        plugins: [QuartoLineHighlight, PdfExport, RevealMenu, QuartoSupport,

          RevealMath,
          RevealNotes,
          RevealSearch,
          RevealZoom
        ]
      });
    </script>
    
    <script>
      // htmlwidgets need to know to resize themselves when slides are shown/hidden.
      // Fire the "slideenter" event (handled by htmlwidgets.js) when the current
      // slide changes (different for each slide format).
      (function () {
        // dispatch for htmlwidgets
        function fireSlideEnter() {
          const event = window.document.createEvent("Event");
          event.initEvent("slideenter", true, true);
          window.document.dispatchEvent(event);
        }

        function fireSlideChanged(previousSlide, currentSlide) {
          fireSlideEnter();

          // dispatch for shiny
          if (window.jQuery) {
            if (previousSlide) {
              window.jQuery(previousSlide).trigger("hidden");
            }
            if (currentSlide) {
              window.jQuery(currentSlide).trigger("shown");
            }
          }
        }

        // hookup for slidy
        if (window.w3c_slidy) {
          window.w3c_slidy.add_observer(function (slide_num) {
            // slide_num starts at position 1
            fireSlideChanged(null, w3c_slidy.slides[slide_num - 1]);
          });
        }

      })();
    </script>

    <script id="quarto-html-after-body" type="application/javascript">
    window.document.addEventListener("DOMContentLoaded", function (event) {
      const toggleBodyColorMode = (bsSheetEl) => {
        const mode = bsSheetEl.getAttribute("data-mode");
        const bodyEl = window.document.querySelector("body");
        if (mode === "dark") {
          bodyEl.classList.add("quarto-dark");
          bodyEl.classList.remove("quarto-light");
        } else {
          bodyEl.classList.add("quarto-light");
          bodyEl.classList.remove("quarto-dark");
        }
      }
      const toggleBodyColorPrimary = () => {
        const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
        if (bsSheetEl) {
          toggleBodyColorMode(bsSheetEl);
        }
      }
      toggleBodyColorPrimary();  
      const tabsets =  window.document.querySelectorAll(".panel-tabset-tabby")
      tabsets.forEach(function(tabset) {
        const tabby = new Tabby('#' + tabset.id);
      });
      const clipboard = new window.ClipboardJS('.code-copy-button', {
        target: function(trigger) {
          return trigger.previousElementSibling;
        }
      });
      clipboard.on('success', function(e) {
        // button target
        const button = e.trigger;
        // don't keep focus
        button.blur();
        // flash "checked"
        button.classList.add('code-copy-button-checked');
        var currentTitle = button.getAttribute("title");
        button.setAttribute("title", "Copied!");
        let tooltip;
        if (window.bootstrap) {
          button.setAttribute("data-bs-toggle", "tooltip");
          button.setAttribute("data-bs-placement", "left");
          button.setAttribute("data-bs-title", "Copied!");
          tooltip = new bootstrap.Tooltip(button, 
            { trigger: "manual", 
              customClass: "code-copy-button-tooltip",
              offset: [0, -8]});
          tooltip.show();    
        }
        setTimeout(function() {
          if (tooltip) {
            tooltip.hide();
            button.removeAttribute("data-bs-title");
            button.removeAttribute("data-bs-toggle");
            button.removeAttribute("data-bs-placement");
          }
          button.setAttribute("title", currentTitle);
          button.classList.remove('code-copy-button-checked');
        }, 1000);
        // clear code selection
        e.clearSelection();
      });
      function tippyHover(el, contentFn) {
        const config = {
          allowHTML: true,
          content: contentFn,
          maxWidth: 500,
          delay: 100,
          arrow: false,
          appendTo: function(el) {
              return el.closest('section.slide') || el.parentElement;
          },
          interactive: true,
          interactiveBorder: 10,
          theme: 'quarto-reveal',
          placement: 'bottom-start'
        };
          config['offset'] = [0,0];
          config['maxWidth'] = 700;
        window.tippy(el, config); 
      }
      const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
      for (var i=0; i<noterefs.length; i++) {
        const ref = noterefs[i];
        tippyHover(ref, function() {
          // use id or data attribute instead here
          let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
          try { href = new URL(href).hash; } catch {}
          const id = href.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          return note.innerHTML;
        });
      }
      const findCites = (el) => {
        const parentEl = el.parentElement;
        if (parentEl) {
          const cites = parentEl.dataset.cites;
          if (cites) {
            return {
              el,
              cites: cites.split(' ')
            };
          } else {
            return findCites(el.parentElement)
          }
        } else {
          return undefined;
        }
      };
      var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
      for (var i=0; i<bibliorefs.length; i++) {
        const ref = bibliorefs[i];
        const citeInfo = findCites(ref);
        if (citeInfo) {
          tippyHover(citeInfo.el, function() {
            var popup = window.document.createElement('div');
            citeInfo.cites.forEach(function(cite) {
              var citeDiv = window.document.createElement('div');
              citeDiv.classList.add('hanging-indent');
              citeDiv.classList.add('csl-entry');
              var biblioDiv = window.document.getElementById('ref-' + cite);
              if (biblioDiv) {
                citeDiv.innerHTML = biblioDiv.innerHTML;
              }
              popup.appendChild(citeDiv);
            });
            return popup.innerHTML;
          });
        }
      }
    });
    </script>
    

</body></html>